In [1]:
import os
import csv
import time
import json
import torch
import random
import argparse
import numpy as np
import pandas as pd
from tqdm.auto import tqdm
from torch.utils.data import DataLoader

from model import DDG_RDE_Network, Codebook
from trainer import CrossValidation, recursive_to
from utils import set_seed, check_dir, eval_skempi_three_modes, save_code, load_config
from dataset import SkempiDataset
from torch.utils.data import random_split
from ipdb import set_trace

/home/hongtan/miniconda3/envs/DDAffinity/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')

In [3]:
train_dataset = SkempiDataset(
            csv_path="/home/hongtan/pretrain_single/dataset/s8716_TEST.csv",
            pdb_dir="/home/hongtan/pretrain_single/final_dataset/pdb",
            cache_dir="/home/hongtan/pretrain_single/final_dataset/train_8026/train_cache",
            patch_size=128
        )

/home/hongtan/miniconda3/envs/DDAffinity/lib/python3.10/site-packages/torch/storage.py:414: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  return torch.load(io.BytesIO(b))


In [4]:
s1657_test_dataset = SkempiDataset(
            csv_path="/home/hongtan/pretrain_single/final_dataset/s669/modified_file.csv",
            pdb_dir="/home/hongtan/pretrain_single/final_dataset/pdb",
            cache_dir="/home/hongtan/pretrain_single/final_dataset/s669/cache",
            patch_size=128
        )

In [5]:
param = json.loads(open("/home/hongtan/pretrain_single/config/param_configs.json", 'r').read())
args = argparse.Namespace(**param)

In [6]:
model = DDG_RDE_Network(args)

In [7]:
import math

import torch
from torch.utils.data import DataLoader
from torch.utils.data._utils.collate import default_collate

DEFAULT_PAD_VALUES = {
    'aa': 21, 
    'chain_nb': -1, 
    'chain_id': ' ', 
}
class PaddingCollate(object):
    def __init__(self, length_ref_key='aa', pad_values=DEFAULT_PAD_VALUES):
        super().__init__()
        self.length_ref_key = length_ref_key
        self.pad_values = pad_values

    @staticmethod
    def _pad_last(x, n, value=0):
        if isinstance(x, torch.Tensor):
            assert x.size(0) <= n
            if x.size(0) == n:
                return x
            pad_size = [n - x.size(0)] + list(x.shape[1:])
            pad = torch.full(pad_size, fill_value=value).to(x)
            return torch.cat([x, pad], dim=0)
        elif isinstance(x, list):
            pad = [value] * (n - len(x))
            return x + pad
        else:
            return x

    @staticmethod
    def _get_common_keys(list_of_dict):
        keys = set(list_of_dict[0].keys())
        for d in list_of_dict[1:]:
            keys = keys.intersection(d.keys())
        return keys

    def _get_pad_value(self, key):
        return self.pad_values.get(key, 0)

    def __call__(self, data_list):
        # 过滤掉 None 的数据
        data_list = [data for data in data_list if data is not None]

        # 如果所有数据都被过滤掉，返回 None
        if not data_list:
            print("Warning: All data in batch are None.")
            return None

        try:
            max_length = max([data[self.length_ref_key].size(0) for data in data_list])
        except KeyError:
            print(f"KeyError: Key '{self.length_ref_key}' not found in one of the data items.")
            return None
        except ValueError:
            print("ValueError: One of the data items is empty.")
            return None

        max_length = math.ceil(max_length / 8) * 8

        exclude_keys = {'esm_embeddings_wt','esm_embeddings_mut'}
        data_list_padded = []
        for data in data_list:
            if data is None:
                continue  # 再次检查并跳过 None 数据

            data_padded = {}
            for k, v in data.items():
                if k in exclude_keys:
                    data_padded[k] = v  # 保持原样，不填充
                else:
                    data_padded[k] = self._pad_last(v, max_length, value=self._get_pad_value(k))
            data_list_padded.append(data_padded)

        # 如果经过填充的数据仍然是空的，返回 None
        if not data_list_padded:
            print("Warning: No valid data to collate after padding.")
            return None

        # 过滤掉任何仍包含 None 的数据
        data_list_padded = [d for d in data_list_padded if all(v is not None for v in d.values())]

        if not data_list_padded:
            print("Warning: All data in batch are invalid after filtering.")
            return None

        batch = default_collate(data_list_padded)
        return batch

dataset_size = len(dataset)
train_size = int(0.9 * dataset_size)  # 80% 用于训练集
val_size = int(0.1 * dataset_size)    # 10% 用于验证集
test_size = dataset_size - train_size - val_size  # 剩下的用于测试集

# 使用 random_split 来划分数据集
train_dataset, val_dataset, test_dataset = random_split(dataset, [train_size, val_size, test_size])
test_dataset = s669_dataset
# 创建 PyGDataLoader 对象
train_loader = DataLoader(train_dataset, batch_size=32, shuffle=True,collate_fn=PaddingCollate(),num_workers=4)
val_loader = DataLoader(val_dataset, batch_size=32, shuffle=True,collate_fn=PaddingCollate(),num_workers=4)
test_loader = DataLoader(test_dataset, batch_size=32, shuffle=True,collate_fn=PaddingCollate(),num_workers=4)

In [8]:
dataset_size = len(train_dataset)
train_size = int(0.9 * dataset_size)  # 80% 用于训练集
val_size = int(0.1 * dataset_size)    # 10% 用于验证集
test_size = dataset_size - train_size - val_size  # 剩下的用于测试集

# 使用 random_split 来划分数据集
train_dataset, val_dataset, test_dataset = random_split(train_dataset, [train_size, val_size, test_size])
test_dataset = s1657_test_dataset
# 创建 PyGDataLoader 对象
train_loader = DataLoader(train_dataset, batch_size=32, shuffle=True,collate_fn=PaddingCollate(),num_workers=4)
val_loader = DataLoader(val_dataset, batch_size=32, shuffle=True,collate_fn=PaddingCollate(),num_workers=4)
test_loader = DataLoader(test_dataset, batch_size=32, shuffle=True,collate_fn=PaddingCollate(),num_workers=4)

In [9]:
print(len(train_dataset))
print(len(test_dataset))

7844
669


In [10]:
vae_model = Codebook(args).to(device)
vae_model.load_state_dict(torch.load("/home/hongtan/pretrain_single/weight/checkpoint/vae_s9683.ckpt", map_location=device))

/tmp/ipykernel_1272984/2080559964.py:2: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  vae_model.load_state_dict(torch.load("/home/hongtan/pretrain_single/weight/checkpoint/v

<All keys matched successfully>

In [11]:
import torch
import numpy as np
from sklearn.metrics import mean_squared_error, r2_score
from scipy.stats import pearsonr
from tqdm.auto import tqdm

# RMSE: Mean Squared Error (MSE), then square root
def compute_rmse(true_vals, pred_vals):
    mse = mean_squared_error(true_vals, pred_vals)
    return np.sqrt(mse)

# PCC: Pearson correlation coefficient
def compute_pcc(true_vals, pred_vals):
    return pearsonr(true_vals, pred_vals)[0]

# R²: Coefficient of determination
def compute_r2(true_vals, pred_vals):
    return r2_score(true_vals, pred_vals)

In [12]:
import torch
import numpy as np
from sklearn.metrics import mean_squared_error, r2_score
from scipy.stats import pearsonr
from tqdm.auto import tqdm

# RMSE: Root Mean Squared Error
def compute_rmse(true_vals, pred_vals):
    mse = mean_squared_error(true_vals, pred_vals)
    return np.sqrt(mse)

# PCC: Pearson correlation coefficient
def compute_pcc(true_vals, pred_vals):
    return pearsonr(true_vals, pred_vals)[0]

# R²: Coefficient of determination (R-squared)
def compute_r2(true_vals, pred_vals):
    return r2_score(true_vals, pred_vals)

import torch
import numpy as np
from sklearn.metrics import mean_squared_error, r2_score
from scipy.stats import pearsonr
from tqdm.auto import tqdm

# RMSE: Root Mean Squared Error
def compute_rmse(true_vals, pred_vals):
    mse = mean_squared_error(true_vals, pred_vals)
    return np.sqrt(mse)

# PCC: Pearson correlation coefficient
def compute_pcc(true_vals, pred_vals):
    return pearsonr(true_vals, pred_vals)[0]

# R²: Coefficient of determination (R-squared)
def compute_r2(true_vals, pred_vals):
    return r2_score(true_vals, pred_vals)

# Helper function to move data to device
def recursive_to(batch, device):
    if isinstance(batch, torch.Tensor):
        return batch.to(device)
    elif isinstance(batch, dict):
        return {k: recursive_to(v, device) for k, v in batch.items()}
    elif isinstance(batch, list):
        return [recursive_to(v, device) for v in batch]
    else:
        return batch

def train_model(model, train_loader, val_loader, test_loader,optimizer, epochs=50, device='cuda:0'):
    model.to(device)
    best_val_loss = float('inf')
    best_model_path = '/home/hongtan/pretrain_single/weight/best_model.pth'
    for epoch in range(epochs):
        model.train()
        running_loss = 0.0

        # Training loop with progress bar
        with tqdm(total=len(train_loader), desc=f'Epoch {epoch+1}/{epochs}', dynamic_ncols=True) as pbar:
            for batch in train_loader:
                batch = recursive_to(batch, device)
                optimizer.zero_grad()

                loss, _ = model(batch,vae_model)
                loss.backward()
                optimizer.step()

                running_loss += loss.item()
                pbar.set_postfix({'Train Loss': loss.item()})
                pbar.update(1)

        avg_train_loss = running_loss / len(train_loader)
        print(f'Epoch {epoch+1}/{epochs}, Train Loss: {avg_train_loss:.4f}')
        
        # Validation loop
        val_rmse, val_pcc, val_r2, avg_val_loss = validate_model(model, val_loader, device)
        print(f'Epoch {epoch+1}/{epochs}, Val Loss: {avg_val_loss:.4f}, Val RMSE: {val_rmse:.4f}, Val PCC: {val_pcc:.4f}, Val R²: {val_r2:.4f}')
        
        test_rmse, test_pcc, test_r2, test_val_loss = test_model(model, test_loader, device)
        print(f'Epoch {epoch+1}/{epochs}, test RMSE: {test_rmse:.4f}, test PCC: {test_pcc:.4f}, test R²: {test_r2:.4f}')
        if avg_val_loss < best_val_loss:
            best_val_loss = avg_val_loss
            torch.save(model.state_dict(), best_model_path)
            print(f"Best model saved with Val Loss: {avg_val_loss:.4f}")

def validate_model(model, val_loader, device):
    model.eval()
    
    all_ddg_true = []
    all_ddg_pred = []
    val_loss = 0.0
    with torch.no_grad():
        for batch in tqdm(val_loader, desc='Validating', dynamic_ncols=True):
            batch = recursive_to(batch, device)
            
            # Forward pass
            loss, output_dict = model(batch,vae_model)
            val_loss += loss.item()

            # Collect true and predicted ddG values
            ddg_true = output_dict['ddG_true'].cpu().numpy()
            ddg_pred = output_dict['ddG_pred'].cpu().numpy()
            
            all_ddg_true.append(ddg_true)
            all_ddg_pred.append(ddg_pred)

    # Flatten lists of ddG values
    all_ddg_true = np.concatenate(all_ddg_true)
    all_ddg_pred = np.concatenate(all_ddg_pred)
    avg_val_loss = val_loss / len(val_loader)
    
    # Compute RMSE, PCC, and R²
    rmse = compute_rmse(all_ddg_true, all_ddg_pred)
    pcc = compute_pcc(all_ddg_true, all_ddg_pred)
    r2 = compute_r2(all_ddg_true, all_ddg_pred)
    
    return rmse, pcc, r2, avg_val_loss

def test_model(model, test_loader, device='cuda:0'):
    model.eval()
    all_ddg_true = []
    all_ddg_pred = []
    test_loss = 0.0
    with torch.no_grad():
        for batch in tqdm(test_loader, desc='Testing', dynamic_ncols=True):
            batch = recursive_to(batch, device)
            
            # Forward pass
            loss, output_dict = model(batch,vae_model)
            test_loss += loss.item()

            # Collect true and predicted ddG values
            ddg_true = output_dict['ddG_true'].cpu().numpy()
            ddg_pred = output_dict['ddG_pred'].cpu().numpy()
            
            all_ddg_true.append(ddg_true)
            all_ddg_pred.append(ddg_pred)

    # Flatten lists of ddG values
    all_ddg_true = np.concatenate(all_ddg_true)
    all_ddg_pred = np.concatenate(all_ddg_pred)
    avg_test_loss = test_loss / len(test_loader)

    # Compute RMSE, PCC, and R²
    rmse = compute_rmse(all_ddg_true, all_ddg_pred)
    pcc = compute_pcc(all_ddg_true, all_ddg_pred)
    r2 = compute_r2(all_ddg_true, all_ddg_pred)
    
    print(f'Test Loss: {avg_test_loss:.4f}, Test RMSE: {rmse:.4f}, Test PCC: {pcc:.4f}, Test R²: {r2:.4f}')
    return rmse, pcc, r2, avg_test_loss


In [13]:
optimizer = torch.optim.Adam(model.parameters(), lr=0.0001, weight_decay=1e-4) #1e-5
scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, factor=0.8, patience=10, min_lr=1e-6)


In [14]:
train_model(model, train_loader, val_loader,test_loader,optimizer = optimizer, epochs=100, device= torch.device('cuda:0'))


Epoch 1/100:   2%|▏         | 4/246 [00:10<06:58,  1.73s/it, Train Loss=12.9]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 1/100:  95%|█████████▍| 233/246 [02:39<00:07,  1.69it/s, Train Loss=6.49] 

Skipping entry 6409 after transform.


Epoch 1/100: 100%|██████████| 246/246 [02:46<00:00,  1.48it/s, Train Loss=1.31]


Epoch 1/100, Train Loss: 4.7167


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.12it/s]


Epoch 1/100, Val Loss: 4.3698, Val RMSE: 2.0974, Val PCC: 0.2044, Val R²: 0.0253


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.04it/s]


Test Loss: 2.6834, Test RMSE: 1.6388, Test PCC: 0.3239, Test R²: -0.0049
Epoch 1/100, test RMSE: 1.6388, test PCC: 0.3239, test R²: -0.0049
Best model saved with Val Loss: 4.3698


Epoch 2/100:  12%|█▏        | 30/246 [00:23<02:17,  1.57it/s, Train Loss=3.05]

Skipping entry 6409 after transform.


Epoch 2/100:  75%|███████▌  | 185/246 [02:11<00:38,  1.58it/s, Train Loss=4.53]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 2/100: 100%|██████████| 246/246 [02:49<00:00,  1.45it/s, Train Loss=0.041]


Epoch 2/100, Train Loss: 4.3585


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.28it/s]


Epoch 2/100, Val Loss: 4.1312, Val RMSE: 1.9889, Val PCC: 0.3643, Val R²: 0.1236


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.67it/s]


Test Loss: 2.6759, Test RMSE: 1.6364, Test PCC: 0.3134, Test R²: -0.0020
Epoch 2/100, test RMSE: 1.6364, test PCC: 0.3134, test R²: -0.0020
Best model saved with Val Loss: 4.1312


Epoch 3/100:  46%|████▌     | 112/246 [01:19<01:23,  1.60it/s, Train Loss=2.1] 

Skipping entry 6409 after transform.


Epoch 3/100:  61%|██████    | 150/246 [01:45<00:54,  1.75it/s, Train Loss=2.06]

Skipping mutation K->S at ('A', '80'): not found in seq_map


Epoch 3/100:  61%|██████▏   | 151/246 [01:45<00:54,  1.76it/s, Train Loss=2.06]

Skipping entry 1703 after transform.


Epoch 3/100: 100%|██████████| 246/246 [02:50<00:00,  1.44it/s, Train Loss=1.17]


Epoch 3/100, Train Loss: 3.9216


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.13it/s]


Epoch 3/100, Val Loss: 3.6605, Val RMSE: 1.9333, Val PCC: 0.4413, Val R²: 0.1719


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.78it/s]


Test Loss: 2.8153, Test RMSE: 1.6771, Test PCC: 0.3348, Test R²: -0.0524
Epoch 3/100, test RMSE: 1.6771, test PCC: 0.3348, test R²: -0.0524
Best model saved with Val Loss: 3.6605


Epoch 4/100:   0%|          | 0/246 [00:00<?, ?it/s]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 4/100:  63%|██████▎   | 154/246 [02:02<00:59,  1.56it/s, Train Loss=2.18]

Skipping entry 6409 after transform.


Epoch 4/100: 100%|██████████| 246/246 [02:58<00:00,  1.38it/s, Train Loss=2.58]


Epoch 4/100, Train Loss: 3.7043


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.18it/s]


Epoch 4/100, Val Loss: 3.5228, Val RMSE: 1.8997, Val PCC: 0.4596, Val R²: 0.2005


Testing: 100%|██████████| 21/21 [00:14<00:00,  1.50it/s]


Test Loss: 2.5301, Test RMSE: 1.5914, Test PCC: 0.3679, Test R²: 0.0523
Epoch 4/100, test RMSE: 1.5914, test PCC: 0.3679, test R²: 0.0523
Best model saved with Val Loss: 3.5228


Epoch 5/100:  31%|███▏      | 77/246 [00:56<01:41,  1.67it/s, Train Loss=5.65] 

Skipping entry 6409 after transform.


Epoch 5/100:  39%|███▉      | 96/246 [01:07<01:25,  1.75it/s, Train Loss=3]   

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 5/100: 100%|██████████| 246/246 [02:33<00:00,  1.60it/s, Train Loss=0.159]


Epoch 5/100, Train Loss: 3.3406


Validating: 100%|██████████| 28/28 [00:18<00:00,  1.55it/s]


Epoch 5/100, Val Loss: 3.5830, Val RMSE: 1.8788, Val PCC: 0.5198, Val R²: 0.2179


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.83it/s]


Test Loss: 2.7300, Test RMSE: 1.6532, Test PCC: 0.3660, Test R²: -0.0226
Epoch 5/100, test RMSE: 1.6532, test PCC: 0.3660, test R²: -0.0226


Epoch 6/100:  10%|█         | 25/246 [00:17<02:18,  1.60it/s, Train Loss=4.01]

Skipping entry 6409 after transform.


Epoch 6/100:  87%|████████▋ | 215/246 [02:13<00:17,  1.76it/s, Train Loss=3]    

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 6/100: 100%|██████████| 246/246 [02:30<00:00,  1.63it/s, Train Loss=1.97]


Epoch 6/100, Train Loss: 2.8189


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.87it/s]


Epoch 6/100, Val Loss: 3.3796, Val RMSE: 1.8420, Val PCC: 0.5328, Val R²: 0.2482


Testing: 100%|██████████| 21/21 [00:08<00:00,  2.42it/s]


Test Loss: 2.6611, Test RMSE: 1.6322, Test PCC: 0.3741, Test R²: 0.0032
Epoch 6/100, test RMSE: 1.6322, test PCC: 0.3741, test R²: 0.0032
Best model saved with Val Loss: 3.3796


Epoch 7/100:  22%|██▏       | 54/246 [00:34<01:58,  1.62it/s, Train Loss=3.95] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 7/100:  42%|████▏     | 104/246 [01:08<01:32,  1.54it/s, Train Loss=4.78]

Skipping entry 6409 after transform.


Epoch 7/100: 100%|██████████| 246/246 [02:40<00:00,  1.54it/s, Train Loss=1.4]  


Epoch 7/100, Train Loss: 2.5247


Validating: 100%|██████████| 28/28 [00:15<00:00,  1.79it/s]


Epoch 7/100, Val Loss: 3.1396, Val RMSE: 1.7494, Val PCC: 0.5950, Val R²: 0.3219


Testing: 100%|██████████| 21/21 [00:17<00:00,  1.22it/s]


Test Loss: 2.5231, Test RMSE: 1.5841, Test PCC: 0.4142, Test R²: 0.0610
Epoch 7/100, test RMSE: 1.5841, test PCC: 0.4142, test R²: 0.0610
Best model saved with Val Loss: 3.1396


Epoch 8/100:  55%|█████▍    | 135/246 [01:36<01:26,  1.28it/s, Train Loss=0.694]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 8/100:  89%|████████▊ | 218/246 [02:30<00:17,  1.60it/s, Train Loss=4.39] 

Skipping entry 6409 after transform.


Epoch 8/100: 100%|██████████| 246/246 [02:46<00:00,  1.48it/s, Train Loss=1.13] 


Epoch 8/100, Train Loss: 2.1969


Validating: 100%|██████████| 28/28 [00:16<00:00,  1.70it/s]


Epoch 8/100, Val Loss: 2.8810, Val RMSE: 1.7082, Val PCC: 0.6103, Val R²: 0.3535


Testing: 100%|██████████| 21/21 [00:23<00:00,  1.11s/it]


Test Loss: 2.4023, Test RMSE: 1.5513, Test PCC: 0.4326, Test R²: 0.0996
Epoch 8/100, test RMSE: 1.5513, test PCC: 0.4326, test R²: 0.0996
Best model saved with Val Loss: 2.8810


Epoch 9/100:  24%|██▎       | 58/246 [00:54<02:37,  1.19it/s, Train Loss=2.02] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 9/100:  41%|████▏     | 102/246 [01:34<03:46,  1.57s/it, Train Loss=0.686]

Skipping entry 6409 after transform.


Epoch 9/100: 100%|██████████| 246/246 [03:13<00:00,  1.27it/s, Train Loss=0.864]


Epoch 9/100, Train Loss: 1.8671


Validating: 100%|██████████| 28/28 [00:26<00:00,  1.06it/s]


Epoch 9/100, Val Loss: 3.0536, Val RMSE: 1.7586, Val PCC: 0.6056, Val R²: 0.3148


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.83it/s]


Test Loss: 2.6820, Test RMSE: 1.6366, Test PCC: 0.4163, Test R²: -0.0022
Epoch 9/100, test RMSE: 1.6366, test PCC: 0.4163, test R²: -0.0022


Epoch 10/100:  11%|█         | 26/246 [00:16<02:07,  1.72it/s, Train Loss=0.422]

Skipping entry 6409 after transform.


Epoch 10/100:  86%|████████▌ | 212/246 [02:08<00:20,  1.69it/s, Train Loss=1.86] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 10/100: 100%|██████████| 246/246 [02:28<00:00,  1.65it/s, Train Loss=0.358]


Epoch 10/100, Train Loss: 1.6536


Validating: 100%|██████████| 28/28 [00:10<00:00,  2.68it/s]


Epoch 10/100, Val Loss: 3.2054, Val RMSE: 1.8004, Val PCC: 0.6038, Val R²: 0.2818


Testing: 100%|██████████| 21/21 [00:10<00:00,  1.93it/s]


Test Loss: 2.8900, Test RMSE: 1.6999, Test PCC: 0.3951, Test R²: -0.0812
Epoch 10/100, test RMSE: 1.6999, test PCC: 0.3951, test R²: -0.0812


Epoch 11/100:  35%|███▌      | 87/246 [01:14<01:38,  1.62it/s, Train Loss=0.866]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 11/100:  86%|████████▌ | 212/246 [02:34<00:19,  1.71it/s, Train Loss=1.11] 

Skipping entry 6409 after transform.


Epoch 11/100: 100%|██████████| 246/246 [02:55<00:00,  1.40it/s, Train Loss=0.141]


Epoch 11/100, Train Loss: 1.4141


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.10it/s]


Epoch 11/100, Val Loss: 2.8320, Val RMSE: 1.6970, Val PCC: 0.6480, Val R²: 0.3619


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.65it/s]


Test Loss: 2.8554, Test RMSE: 1.6895, Test PCC: 0.3843, Test R²: -0.0680
Epoch 11/100, test RMSE: 1.6895, test PCC: 0.3843, test R²: -0.0680
Best model saved with Val Loss: 2.8320


Epoch 12/100:  63%|██████▎   | 155/246 [01:42<00:55,  1.63it/s, Train Loss=1.23] 

Skipping entry 6409 after transform.


Epoch 12/100:  92%|█████████▏| 227/246 [02:24<00:10,  1.74it/s, Train Loss=1.13] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 12/100: 100%|██████████| 246/246 [02:35<00:00,  1.58it/s, Train Loss=1.85] 


Epoch 12/100, Train Loss: 1.1776


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.16it/s]


Epoch 12/100, Val Loss: 2.8569, Val RMSE: 1.6939, Val PCC: 0.6360, Val R²: 0.3643


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.88it/s]


Test Loss: 2.7022, Test RMSE: 1.6421, Test PCC: 0.3997, Test R²: -0.0090
Epoch 12/100, test RMSE: 1.6421, test PCC: 0.3997, test R²: -0.0090


Epoch 13/100:  30%|██▉       | 73/246 [00:50<01:40,  1.73it/s, Train Loss=0.67] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 13/100:  56%|█████▌    | 138/246 [01:30<01:02,  1.72it/s, Train Loss=0.646]

Skipping entry 6409 after transform.


Epoch 13/100: 100%|██████████| 246/246 [02:32<00:00,  1.61it/s, Train Loss=0.363]


Epoch 13/100, Train Loss: 1.0092


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.30it/s]


Epoch 13/100, Val Loss: 2.7210, Val RMSE: 1.6646, Val PCC: 0.6487, Val R²: 0.3861


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.22it/s]


Test Loss: 2.7347, Test RMSE: 1.6533, Test PCC: 0.3883, Test R²: -0.0228
Epoch 13/100, test RMSE: 1.6533, test PCC: 0.3883, test R²: -0.0228
Best model saved with Val Loss: 2.7210


Epoch 14/100:  64%|██████▍   | 157/246 [01:41<00:51,  1.73it/s, Train Loss=0.771]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 14/100:  86%|████████▌ | 212/246 [02:14<00:19,  1.71it/s, Train Loss=0.684]

Skipping entry 6409 after transform.


Epoch 14/100: 100%|██████████| 246/246 [02:34<00:00,  1.59it/s, Train Loss=0.589]


Epoch 14/100, Train Loss: 0.9607


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.92it/s]


Epoch 14/100, Val Loss: 2.8329, Val RMSE: 1.6871, Val PCC: 0.6485, Val R²: 0.3694


Testing: 100%|██████████| 21/21 [00:13<00:00,  1.54it/s]


Test Loss: 2.7421, Test RMSE: 1.6536, Test PCC: 0.4091, Test R²: -0.0231
Epoch 14/100, test RMSE: 1.6536, test PCC: 0.4091, test R²: -0.0231


Epoch 15/100:   9%|▊         | 21/246 [00:20<03:17,  1.14it/s, Train Loss=1.09] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 15/100:  61%|██████    | 150/246 [01:45<00:55,  1.73it/s, Train Loss=0.721]

Skipping entry 6409 after transform.


Epoch 15/100: 100%|██████████| 246/246 [02:51<00:00,  1.44it/s, Train Loss=1.97] 


Epoch 15/100, Train Loss: 0.8409


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.17it/s]


Epoch 15/100, Val Loss: 2.7496, Val RMSE: 1.6667, Val PCC: 0.6483, Val R²: 0.3845


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.87it/s]


Test Loss: 2.5857, Test RMSE: 1.6071, Test PCC: 0.4222, Test R²: 0.0336
Epoch 15/100, test RMSE: 1.6071, test PCC: 0.4222, test R²: 0.0336


Epoch 16/100:  60%|█████▉    | 147/246 [01:31<00:58,  1.70it/s, Train Loss=0.988]

Skipping entry 6409 after transform.


Epoch 16/100:  93%|█████████▎| 229/246 [02:27<00:12,  1.41it/s, Train Loss=0.835]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 16/100: 100%|██████████| 246/246 [02:39<00:00,  1.55it/s, Train Loss=0.702]


Epoch 16/100, Train Loss: 0.7476


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.26it/s]


Epoch 16/100, Val Loss: 3.0455, Val RMSE: 1.7605, Val PCC: 0.6291, Val R²: 0.3133


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.71it/s]


Test Loss: 3.1111, Test RMSE: 1.7656, Test PCC: 0.3484, Test R²: -0.1664
Epoch 16/100, test RMSE: 1.7656, test PCC: 0.3484, test R²: -0.1664


Epoch 17/100:  24%|██▍       | 59/246 [00:39<01:49,  1.71it/s, Train Loss=0.339]

Skipping entry 6409 after transform.


Epoch 17/100:  83%|████████▎ | 203/246 [02:05<00:25,  1.71it/s, Train Loss=0.514]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 17/100: 100%|██████████| 246/246 [02:29<00:00,  1.64it/s, Train Loss=0.401]


Epoch 17/100, Train Loss: 0.6205


Validating: 100%|██████████| 28/28 [00:15<00:00,  1.78it/s]


Epoch 17/100, Val Loss: 2.6191, Val RMSE: 1.6318, Val PCC: 0.6650, Val R²: 0.4101


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.01it/s]


Test Loss: 2.5688, Test RMSE: 1.6027, Test PCC: 0.3983, Test R²: 0.0389
Epoch 17/100, test RMSE: 1.6027, test PCC: 0.3983, test R²: 0.0389
Best model saved with Val Loss: 2.6191


Epoch 18/100:  30%|███       | 75/246 [00:51<01:47,  1.59it/s, Train Loss=0.305]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 18/100:  77%|███████▋  | 190/246 [02:00<00:33,  1.66it/s, Train Loss=0.299]

Skipping entry 6409 after transform.


Epoch 18/100: 100%|██████████| 246/246 [02:32<00:00,  1.61it/s, Train Loss=0.175]


Epoch 18/100, Train Loss: 0.5389


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.29it/s]


Epoch 18/100, Val Loss: 2.9763, Val RMSE: 1.7405, Val PCC: 0.6219, Val R²: 0.3288


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.86it/s]


Test Loss: 2.6350, Test RMSE: 1.6240, Test PCC: 0.4003, Test R²: 0.0131
Epoch 18/100, test RMSE: 1.6240, test PCC: 0.4003, test R²: 0.0131


Epoch 19/100:  44%|████▍     | 109/246 [01:20<01:20,  1.71it/s, Train Loss=0.672]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 19/100:  86%|████████▌ | 212/246 [02:19<00:19,  1.72it/s, Train Loss=0.44] 

Skipping entry 6409 after transform.


Epoch 19/100: 100%|██████████| 246/246 [02:38<00:00,  1.55it/s, Train Loss=0.115]


Epoch 19/100, Train Loss: 0.4590


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.21it/s]


Epoch 19/100, Val Loss: 2.7953, Val RMSE: 1.6898, Val PCC: 0.6439, Val R²: 0.3674


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.63it/s]


Test Loss: 2.5442, Test RMSE: 1.5960, Test PCC: 0.4239, Test R²: 0.0469
Epoch 19/100, test RMSE: 1.5960, test PCC: 0.4239, test R²: 0.0469


Epoch 20/100:   0%|          | 1/246 [00:01<06:45,  1.66s/it, Train Loss=0.476]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 20/100:  89%|████████▉ | 219/246 [02:11<00:15,  1.76it/s, Train Loss=0.798]

Skipping entry 6409 after transform.


Epoch 20/100: 100%|██████████| 246/246 [02:26<00:00,  1.68it/s, Train Loss=0.297]


Epoch 20/100, Train Loss: 0.4334


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.31it/s]


Epoch 20/100, Val Loss: 3.2797, Val RMSE: 1.7046, Val PCC: 0.6299, Val R²: 0.3562


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.76it/s]


Test Loss: 2.6123, Test RMSE: 1.6150, Test PCC: 0.4039, Test R²: 0.0241
Epoch 20/100, test RMSE: 1.6150, test PCC: 0.4039, test R²: 0.0241


Epoch 21/100:  41%|████▏     | 102/246 [01:12<01:24,  1.70it/s, Train Loss=0.299]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 21/100:  68%|██████▊   | 167/246 [01:51<00:50,  1.57it/s, Train Loss=0.219]

Skipping entry 6409 after transform.


Epoch 21/100: 100%|██████████| 246/246 [02:40<00:00,  1.53it/s, Train Loss=0.183]


Epoch 21/100, Train Loss: 0.4009


Validating: 100%|██████████| 28/28 [00:19<00:00,  1.46it/s]


Epoch 21/100, Val Loss: 2.8059, Val RMSE: 1.6899, Val PCC: 0.6417, Val R²: 0.3672


Testing: 100%|██████████| 21/21 [00:14<00:00,  1.49it/s]


Test Loss: 2.6911, Test RMSE: 1.6393, Test PCC: 0.3927, Test R²: -0.0056
Epoch 21/100, test RMSE: 1.6393, test PCC: 0.3927, test R²: -0.0056


Epoch 22/100:  56%|█████▌    | 138/246 [01:25<01:01,  1.74it/s, Train Loss=0.576] 

Skipping entry 6409 after transform.


Epoch 22/100:  94%|█████████▍| 232/246 [02:19<00:08,  1.75it/s, Train Loss=0.284]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 22/100: 100%|██████████| 246/246 [02:27<00:00,  1.67it/s, Train Loss=0.271]


Epoch 22/100, Train Loss: 0.3775


Validating: 100%|██████████| 28/28 [00:17<00:00,  1.59it/s]


Epoch 22/100, Val Loss: 2.9325, Val RMSE: 1.7220, Val PCC: 0.6328, Val R²: 0.3430


Testing: 100%|██████████| 21/21 [00:16<00:00,  1.29it/s]


Test Loss: 2.7021, Test RMSE: 1.6393, Test PCC: 0.3912, Test R²: -0.0056
Epoch 22/100, test RMSE: 1.6393, test PCC: 0.3912, test R²: -0.0056


Epoch 23/100:  42%|████▏     | 103/246 [01:30<02:06,  1.13it/s, Train Loss=0.434]

Skipping mutation K->S at ('A', '80'): not found in seq_map


Epoch 23/100:  42%|████▏     | 104/246 [01:30<01:54,  1.24it/s, Train Loss=0.134]

Skipping entry 1703 after transform.


Epoch 23/100:  87%|████████▋ | 214/246 [02:43<00:20,  1.56it/s, Train Loss=0.554]

Skipping entry 6409 after transform.


Epoch 23/100: 100%|██████████| 246/246 [03:02<00:00,  1.35it/s, Train Loss=0.54] 


Epoch 23/100, Train Loss: 0.3513


Validating: 100%|██████████| 28/28 [00:20<00:00,  1.35it/s]


Epoch 23/100, Val Loss: 3.1447, Val RMSE: 1.7620, Val PCC: 0.6086, Val R²: 0.3121


Testing: 100%|██████████| 21/21 [00:18<00:00,  1.14it/s]


Test Loss: 2.5600, Test RMSE: 1.5987, Test PCC: 0.4189, Test R²: 0.0437
Epoch 23/100, test RMSE: 1.5987, test PCC: 0.4189, test R²: 0.0437


Epoch 24/100:  15%|█▌        | 38/246 [00:29<02:03,  1.68it/s, Train Loss=0.625]

Skipping entry 6409 after transform.


Epoch 24/100:  54%|█████▍    | 134/246 [01:26<01:21,  1.38it/s, Train Loss=0.295]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 24/100: 100%|██████████| 246/246 [02:36<00:00,  1.57it/s, Train Loss=0.397]


Epoch 24/100, Train Loss: 0.3609


Validating: 100%|██████████| 28/28 [00:26<00:00,  1.06it/s]


Epoch 24/100, Val Loss: 2.8067, Val RMSE: 1.6942, Val PCC: 0.6487, Val R²: 0.3640


Testing: 100%|██████████| 21/21 [00:16<00:00,  1.29it/s]


Test Loss: 2.5536, Test RMSE: 1.5971, Test PCC: 0.4107, Test R²: 0.0456
Epoch 24/100, test RMSE: 1.5971, test PCC: 0.4107, test R²: 0.0456


Epoch 25/100:  27%|██▋       | 67/246 [00:46<02:09,  1.38it/s, Train Loss=0.262]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 25/100:  38%|███▊      | 93/246 [01:04<01:29,  1.71it/s, Train Loss=0.27] 

Skipping entry 6409 after transform.


Epoch 25/100: 100%|██████████| 246/246 [02:54<00:00,  1.41it/s, Train Loss=0.235]


Epoch 25/100, Train Loss: 0.3213


Validating: 100%|██████████| 28/28 [00:17<00:00,  1.64it/s]


Epoch 25/100, Val Loss: 2.8853, Val RMSE: 1.6918, Val PCC: 0.6430, Val R²: 0.3659


Testing: 100%|██████████| 21/21 [00:19<00:00,  1.07it/s]


Test Loss: 2.5434, Test RMSE: 1.5929, Test PCC: 0.4199, Test R²: 0.0505
Epoch 25/100, test RMSE: 1.5929, test PCC: 0.4199, test R²: 0.0505


Epoch 26/100:  35%|███▍      | 86/246 [01:18<01:37,  1.64it/s, Train Loss=0.275] 

Skipping entry 6409 after transform.


Epoch 26/100:  67%|██████▋   | 166/246 [02:11<00:52,  1.51it/s, Train Loss=0.376]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 26/100: 100%|██████████| 246/246 [03:00<00:00,  1.36it/s, Train Loss=0.257]


Epoch 26/100, Train Loss: 0.2861


Validating: 100%|██████████| 28/28 [00:20<00:00,  1.37it/s]


Epoch 26/100, Val Loss: 2.7847, Val RMSE: 1.6710, Val PCC: 0.6460, Val R²: 0.3813


Testing: 100%|██████████| 21/21 [00:10<00:00,  1.95it/s]


Test Loss: 2.4372, Test RMSE: 1.5597, Test PCC: 0.4194, Test R²: 0.0898
Epoch 26/100, test RMSE: 1.5597, test PCC: 0.4194, test R²: 0.0898


Epoch 27/100:  19%|█▉        | 47/246 [00:42<01:57,  1.69it/s, Train Loss=0.512]

Skipping entry 6409 after transform.


Epoch 27/100:  31%|███       | 76/246 [01:09<02:06,  1.35it/s, Train Loss=0.202]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 27/100: 100%|██████████| 246/246 [03:03<00:00,  1.34it/s, Train Loss=0.296]


Epoch 27/100, Train Loss: 0.2633


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.92it/s]


Epoch 27/100, Val Loss: 2.7627, Val RMSE: 1.6700, Val PCC: 0.6453, Val R²: 0.3821


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.83it/s]


Test Loss: 2.4924, Test RMSE: 1.5792, Test PCC: 0.4085, Test R²: 0.0668
Epoch 27/100, test RMSE: 1.5792, test PCC: 0.4085, test R²: 0.0668


Epoch 28/100:  10%|▉         | 24/246 [00:19<02:30,  1.47it/s, Train Loss=0.46] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 28/100:  78%|███████▊  | 191/246 [02:20<00:44,  1.23it/s, Train Loss=0.323]

Skipping entry 6409 after transform.


Epoch 28/100: 100%|██████████| 246/246 [02:54<00:00,  1.41it/s, Train Loss=0.171]


Epoch 28/100, Train Loss: 0.2497


Validating: 100%|██████████| 28/28 [00:15<00:00,  1.78it/s]


Epoch 28/100, Val Loss: 2.9098, Val RMSE: 1.7049, Val PCC: 0.6393, Val R²: 0.3560


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.17it/s]


Test Loss: 2.5405, Test RMSE: 1.5927, Test PCC: 0.3933, Test R²: 0.0508
Epoch 28/100, test RMSE: 1.5927, test PCC: 0.3933, test R²: 0.0508


Epoch 29/100:   2%|▏         | 4/246 [00:07<05:45,  1.43s/it, Train Loss=0.157]

Skipping entry 6409 after transform.


Epoch 29/100:  46%|████▌     | 112/246 [01:40<01:44,  1.28it/s, Train Loss=0.303]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 29/100: 100%|██████████| 246/246 [03:27<00:00,  1.18it/s, Train Loss=0.302] 


Epoch 29/100, Train Loss: 0.2261


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.99it/s]


Epoch 29/100, Val Loss: 2.7899, Val RMSE: 1.6844, Val PCC: 0.6403, Val R²: 0.3714


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.65it/s]


Test Loss: 2.5390, Test RMSE: 1.5946, Test PCC: 0.3972, Test R²: 0.0486
Epoch 29/100, test RMSE: 1.5946, test PCC: 0.3972, test R²: 0.0486


Epoch 30/100:  13%|█▎        | 32/246 [00:29<02:41,  1.32it/s, Train Loss=0.294]

Skipping entry 6409 after transform.


Epoch 30/100:  44%|████▍     | 108/246 [01:26<02:09,  1.07it/s, Train Loss=0.177]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 30/100: 100%|██████████| 246/246 [02:50<00:00,  1.44it/s, Train Loss=0.104] 


Epoch 30/100, Train Loss: 0.2340


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.02it/s]


Epoch 30/100, Val Loss: 2.9526, Val RMSE: 1.6896, Val PCC: 0.6373, Val R²: 0.3675


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.62it/s]


Test Loss: 2.5499, Test RMSE: 1.5957, Test PCC: 0.3892, Test R²: 0.0472
Epoch 30/100, test RMSE: 1.5957, test PCC: 0.3892, test R²: 0.0472


Epoch 31/100:  22%|██▏       | 55/246 [00:39<01:50,  1.73it/s, Train Loss=0.501] 

Skipping entry 6409 after transform.


Epoch 31/100:  73%|███████▎  | 180/246 [01:54<00:37,  1.76it/s, Train Loss=0.297]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 31/100: 100%|██████████| 246/246 [02:32<00:00,  1.62it/s, Train Loss=0.0701]


Epoch 31/100, Train Loss: 0.1996


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.12it/s]


Epoch 31/100, Val Loss: 2.8614, Val RMSE: 1.6555, Val PCC: 0.6564, Val R²: 0.3927


Testing: 100%|██████████| 21/21 [00:15<00:00,  1.38it/s]


Test Loss: 2.6609, Test RMSE: 1.6336, Test PCC: 0.3715, Test R²: 0.0014
Epoch 31/100, test RMSE: 1.6336, test PCC: 0.3715, test R²: 0.0014


Epoch 32/100:  57%|█████▋    | 139/246 [01:36<01:04,  1.67it/s, Train Loss=0.115] 

Skipping entry 6409 after transform.


Epoch 32/100:  64%|██████▍   | 157/246 [01:47<00:52,  1.70it/s, Train Loss=0.225] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 32/100: 100%|██████████| 246/246 [02:44<00:00,  1.50it/s, Train Loss=0.249] 


Epoch 32/100, Train Loss: 0.1762


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.22it/s]


Epoch 32/100, Val Loss: 2.7061, Val RMSE: 1.6590, Val PCC: 0.6498, Val R²: 0.3902


Testing: 100%|██████████| 21/21 [00:13<00:00,  1.51it/s]


Test Loss: 2.5774, Test RMSE: 1.6032, Test PCC: 0.3901, Test R²: 0.0383
Epoch 32/100, test RMSE: 1.6032, test PCC: 0.3901, test R²: 0.0383


Epoch 33/100:   7%|▋         | 16/246 [00:12<02:26,  1.57it/s, Train Loss=0.117] 

Skipping entry 6409 after transform.


Epoch 33/100:  89%|████████▉ | 220/246 [02:47<00:16,  1.55it/s, Train Loss=0.112] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 33/100: 100%|██████████| 246/246 [03:05<00:00,  1.33it/s, Train Loss=0.276] 


Epoch 33/100, Train Loss: 0.1817


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.14it/s]


Epoch 33/100, Val Loss: 2.7094, Val RMSE: 1.6610, Val PCC: 0.6480, Val R²: 0.3887


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.00it/s]


Test Loss: 2.4200, Test RMSE: 1.5526, Test PCC: 0.4184, Test R²: 0.0980
Epoch 33/100, test RMSE: 1.5526, test PCC: 0.4184, test R²: 0.0980


Epoch 34/100:  52%|█████▏    | 127/246 [01:23<01:13,  1.62it/s, Train Loss=0.201] 

Skipping entry 6409 after transform.


Epoch 34/100:  63%|██████▎   | 155/246 [01:40<00:55,  1.65it/s, Train Loss=0.132] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 34/100: 100%|██████████| 246/246 [02:50<00:00,  1.44it/s, Train Loss=0.0692]


Epoch 34/100, Train Loss: 0.1807


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.15it/s]


Epoch 34/100, Val Loss: 2.6189, Val RMSE: 1.6217, Val PCC: 0.6684, Val R²: 0.4173


Testing: 100%|██████████| 21/21 [00:10<00:00,  1.91it/s]


Test Loss: 2.4226, Test RMSE: 1.5582, Test PCC: 0.4163, Test R²: 0.0915
Epoch 34/100, test RMSE: 1.5582, test PCC: 0.4163, test R²: 0.0915
Best model saved with Val Loss: 2.6189


Epoch 35/100:  38%|███▊      | 94/246 [01:21<01:34,  1.61it/s, Train Loss=0.158] 

Skipping entry 6409 after transform.


Epoch 35/100:  86%|████████▌ | 211/246 [02:45<00:21,  1.59it/s, Train Loss=0.626] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 35/100: 100%|██████████| 246/246 [03:19<00:00,  1.23it/s, Train Loss=0.997] 


Epoch 35/100, Train Loss: 0.1824


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.15it/s]


Epoch 35/100, Val Loss: 2.5904, Val RMSE: 1.6210, Val PCC: 0.6620, Val R²: 0.4178


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.09it/s]


Test Loss: 2.4510, Test RMSE: 1.5653, Test PCC: 0.3963, Test R²: 0.0832
Epoch 35/100, test RMSE: 1.5653, test PCC: 0.3963, test R²: 0.0832
Best model saved with Val Loss: 2.5904


Epoch 36/100:  69%|██████▉   | 170/246 [02:57<00:47,  1.60it/s, Train Loss=0.152]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 36/100:  79%|███████▉  | 195/246 [03:14<00:47,  1.07it/s, Train Loss=0.128]

Skipping entry 6409 after transform.


Epoch 36/100: 100%|██████████| 246/246 [03:59<00:00,  1.03it/s, Train Loss=0.0239]


Epoch 36/100, Train Loss: 0.2937


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.14it/s]


Epoch 36/100, Val Loss: 2.8908, Val RMSE: 1.7144, Val PCC: 0.6224, Val R²: 0.3488


Testing: 100%|██████████| 21/21 [00:15<00:00,  1.36it/s]


Test Loss: 2.4649, Test RMSE: 1.5658, Test PCC: 0.4095, Test R²: 0.0826
Epoch 36/100, test RMSE: 1.5658, test PCC: 0.4095, test R²: 0.0826


Epoch 37/100:   0%|          | 0/246 [00:00<?, ?it/s]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 37/100:  63%|██████▎   | 156/246 [01:37<00:54,  1.65it/s, Train Loss=0.199] 

Skipping entry 6409 after transform.


Epoch 37/100: 100%|██████████| 246/246 [02:30<00:00,  1.63it/s, Train Loss=0.155] 


Epoch 37/100, Train Loss: 0.2086


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.97it/s]


Epoch 37/100, Val Loss: 2.8792, Val RMSE: 1.6654, Val PCC: 0.6506, Val R²: 0.3855


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.01it/s]


Test Loss: 2.5526, Test RMSE: 1.5983, Test PCC: 0.3660, Test R²: 0.0441
Epoch 37/100, test RMSE: 1.5983, test PCC: 0.3660, test R²: 0.0441


Epoch 38/100:  30%|███       | 75/246 [00:50<01:44,  1.64it/s, Train Loss=0.146] 

Skipping entry 6409 after transform.


Epoch 38/100:  33%|███▎      | 81/246 [00:53<01:39,  1.65it/s, Train Loss=0.109] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 38/100: 100%|██████████| 246/246 [02:32<00:00,  1.61it/s, Train Loss=0.0424]


Epoch 38/100, Train Loss: 0.1575


Validating: 100%|██████████| 28/28 [00:16<00:00,  1.74it/s]


Epoch 38/100, Val Loss: 2.6332, Val RMSE: 1.6407, Val PCC: 0.6555, Val R²: 0.4036


Testing: 100%|██████████| 21/21 [00:15<00:00,  1.39it/s]


Test Loss: 2.3599, Test RMSE: 1.5364, Test PCC: 0.4267, Test R²: 0.1167
Epoch 38/100, test RMSE: 1.5364, test PCC: 0.4267, test R²: 0.1167


Epoch 39/100:   4%|▎         | 9/246 [00:11<03:23,  1.16it/s, Train Loss=0.253] 

Skipping entry 6409 after transform.


Epoch 39/100:  31%|███▏      | 77/246 [00:53<01:37,  1.74it/s, Train Loss=0.103] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 39/100: 100%|██████████| 246/246 [02:35<00:00,  1.58it/s, Train Loss=0.0527]


Epoch 39/100, Train Loss: 0.1430


Validating: 100%|██████████| 28/28 [00:10<00:00,  2.62it/s]


Epoch 39/100, Val Loss: 2.8441, Val RMSE: 1.6777, Val PCC: 0.6399, Val R²: 0.3764


Testing: 100%|██████████| 21/21 [00:16<00:00,  1.24it/s]


Test Loss: 2.3624, Test RMSE: 1.5366, Test PCC: 0.4260, Test R²: 0.1166
Epoch 39/100, test RMSE: 1.5366, test PCC: 0.4260, test R²: 0.1166


Epoch 40/100:  13%|█▎        | 33/246 [00:23<02:04,  1.72it/s, Train Loss=0.0408]

Skipping entry 6409 after transform.


Epoch 40/100:  37%|███▋      | 91/246 [00:57<01:30,  1.72it/s, Train Loss=0.0715]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 40/100: 100%|██████████| 246/246 [02:29<00:00,  1.64it/s, Train Loss=0.279] 


Epoch 40/100, Train Loss: 0.1193


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.01it/s]


Epoch 40/100, Val Loss: 2.7099, Val RMSE: 1.6318, Val PCC: 0.6620, Val R²: 0.4100


Testing: 100%|██████████| 21/21 [00:10<00:00,  1.96it/s]


Test Loss: 2.4751, Test RMSE: 1.5715, Test PCC: 0.3955, Test R²: 0.0759
Epoch 40/100, test RMSE: 1.5715, test PCC: 0.3955, test R²: 0.0759


Epoch 41/100:  54%|█████▎    | 132/246 [01:54<01:18,  1.45it/s, Train Loss=0.189] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 41/100:  73%|███████▎  | 179/246 [02:33<00:50,  1.32it/s, Train Loss=0.116] 

Skipping entry 6409 after transform.


Epoch 41/100: 100%|██████████| 246/246 [03:25<00:00,  1.20it/s, Train Loss=0.116] 


Epoch 41/100, Train Loss: 0.1223


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.30it/s]


Epoch 41/100, Val Loss: 2.7687, Val RMSE: 1.6791, Val PCC: 0.6370, Val R²: 0.3753


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.73it/s]


Test Loss: 2.3831, Test RMSE: 1.5437, Test PCC: 0.4193, Test R²: 0.1083
Epoch 41/100, test RMSE: 1.5437, test PCC: 0.4193, test R²: 0.1083


Epoch 42/100:  82%|████████▏ | 201/246 [02:15<00:45,  1.00s/it, Train Loss=0.102] 

Skipping entry 6409 after transform.


Epoch 42/100:  90%|█████████ | 222/246 [02:30<00:16,  1.49it/s, Train Loss=0.146] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 42/100: 100%|██████████| 246/246 [02:45<00:00,  1.49it/s, Train Loss=0.103] 


Epoch 42/100, Train Loss: 0.1287


Validating: 100%|██████████| 28/28 [00:15<00:00,  1.80it/s]


Epoch 42/100, Val Loss: 2.7153, Val RMSE: 1.6527, Val PCC: 0.6492, Val R²: 0.3948


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.03it/s]


Test Loss: 2.4074, Test RMSE: 1.5504, Test PCC: 0.4151, Test R²: 0.1006
Epoch 42/100, test RMSE: 1.5504, test PCC: 0.4151, test R²: 0.1006


Epoch 43/100:  35%|███▌      | 87/246 [00:54<01:32,  1.71it/s, Train Loss=0.0593]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 43/100:  67%|██████▋   | 165/246 [01:41<00:46,  1.73it/s, Train Loss=0.0962]

Skipping entry 6409 after transform.


Epoch 43/100: 100%|██████████| 246/246 [02:27<00:00,  1.66it/s, Train Loss=0.0827]


Epoch 43/100, Train Loss: 0.1395


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.40it/s]


Epoch 43/100, Val Loss: 2.7162, Val RMSE: 1.6589, Val PCC: 0.6507, Val R²: 0.3902


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.10it/s]


Test Loss: 2.3052, Test RMSE: 1.5183, Test PCC: 0.4497, Test R²: 0.1374
Epoch 43/100, test RMSE: 1.5183, test PCC: 0.4497, test R²: 0.1374


Epoch 44/100:   9%|▉         | 22/246 [00:15<02:14,  1.67it/s, Train Loss=0.111] 

Skipping entry 6409 after transform.


Epoch 44/100:  67%|██████▋   | 165/246 [01:40<00:47,  1.72it/s, Train Loss=0.0479]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 44/100: 100%|██████████| 246/246 [02:27<00:00,  1.67it/s, Train Loss=0.033] 


Epoch 44/100, Train Loss: 0.1399


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.44it/s]


Epoch 44/100, Val Loss: 3.2091, Val RMSE: 1.6895, Val PCC: 0.6386, Val R²: 0.3676


Testing: 100%|██████████| 21/21 [00:13<00:00,  1.53it/s]


Test Loss: 2.4012, Test RMSE: 1.5492, Test PCC: 0.4214, Test R²: 0.1020
Epoch 44/100, test RMSE: 1.5492, test PCC: 0.4214, test R²: 0.1020


Epoch 45/100:  57%|█████▋    | 141/246 [01:25<01:01,  1.70it/s, Train Loss=0.108] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 45/100:  76%|███████▌  | 187/246 [01:55<00:33,  1.75it/s, Train Loss=0.249] 

Skipping entry 6409 after transform.


Epoch 45/100: 100%|██████████| 246/246 [02:31<00:00,  1.63it/s, Train Loss=0.128] 


Epoch 45/100, Train Loss: 0.1539


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.45it/s]


Epoch 45/100, Val Loss: 2.7434, Val RMSE: 1.6660, Val PCC: 0.6498, Val R²: 0.3851


Testing: 100%|██████████| 21/21 [00:08<00:00,  2.37it/s]


Test Loss: 2.3608, Test RMSE: 1.5372, Test PCC: 0.4254, Test R²: 0.1158
Epoch 45/100, test RMSE: 1.5372, test PCC: 0.4254, test R²: 0.1158


Epoch 46/100:  47%|████▋     | 115/246 [01:31<01:18,  1.67it/s, Train Loss=0.162] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 46/100:  56%|█████▌    | 137/246 [01:46<01:06,  1.65it/s, Train Loss=0.109] 

Skipping entry 6409 after transform.


Epoch 46/100: 100%|██████████| 246/246 [02:55<00:00,  1.40it/s, Train Loss=0.00352]


Epoch 46/100, Train Loss: 0.1560


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.25it/s]


Epoch 46/100, Val Loss: 2.6671, Val RMSE: 1.6520, Val PCC: 0.6476, Val R²: 0.3953


Testing: 100%|██████████| 21/21 [00:08<00:00,  2.45it/s]


Test Loss: 2.2881, Test RMSE: 1.5115, Test PCC: 0.4350, Test R²: 0.1452
Epoch 46/100, test RMSE: 1.5115, test PCC: 0.4350, test R²: 0.1452


Epoch 47/100:  20%|█▉        | 48/246 [00:30<01:56,  1.70it/s, Train Loss=0.151] 

Skipping entry 6409 after transform.


Epoch 47/100:  62%|██████▏   | 153/246 [01:39<00:53,  1.73it/s, Train Loss=0.138] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 47/100: 100%|██████████| 246/246 [02:38<00:00,  1.55it/s, Train Loss=0.0444]


Epoch 47/100, Train Loss: 0.1334


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.48it/s]


Epoch 47/100, Val Loss: 2.7774, Val RMSE: 1.6685, Val PCC: 0.6420, Val R²: 0.3832


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.11it/s]


Test Loss: 2.3040, Test RMSE: 1.5170, Test PCC: 0.4452, Test R²: 0.1390
Epoch 47/100, test RMSE: 1.5170, test PCC: 0.4452, test R²: 0.1390


Epoch 48/100:  34%|███▍      | 84/246 [00:51<01:33,  1.74it/s, Train Loss=0.132] 

Skipping entry 6409 after transform.


Epoch 48/100:  35%|███▍      | 85/246 [00:51<01:32,  1.74it/s, Train Loss=0.16] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 48/100: 100%|██████████| 246/246 [02:26<00:00,  1.68it/s, Train Loss=0.0909]


Epoch 48/100, Train Loss: 0.1209


Validating: 100%|██████████| 28/28 [00:15<00:00,  1.76it/s]


Epoch 48/100, Val Loss: 2.7265, Val RMSE: 1.6530, Val PCC: 0.6490, Val R²: 0.3946


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.83it/s]


Test Loss: 2.3448, Test RMSE: 1.5326, Test PCC: 0.4226, Test R²: 0.1211
Epoch 48/100, test RMSE: 1.5326, test PCC: 0.4226, test R²: 0.1211


Epoch 49/100:  80%|████████  | 197/246 [01:57<00:27,  1.76it/s, Train Loss=0.0955]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 49/100:  90%|████████▉ | 221/246 [02:10<00:14,  1.75it/s, Train Loss=0.0875]

Skipping entry 6409 after transform.


Epoch 49/100: 100%|██████████| 246/246 [02:26<00:00,  1.67it/s, Train Loss=0.129] 


Epoch 49/100, Train Loss: 0.1204


Validating: 100%|██████████| 28/28 [00:17<00:00,  1.64it/s]


Epoch 49/100, Val Loss: 2.6998, Val RMSE: 1.6598, Val PCC: 0.6502, Val R²: 0.3896


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.29it/s]


Test Loss: 2.2881, Test RMSE: 1.5112, Test PCC: 0.4339, Test R²: 0.1455
Epoch 49/100, test RMSE: 1.5112, test PCC: 0.4339, test R²: 0.1455


Epoch 50/100:  80%|████████  | 198/246 [02:06<00:28,  1.70it/s, Train Loss=0.0935]

Skipping entry 6409 after transform.


Epoch 50/100:  92%|█████████▏| 226/246 [02:22<00:11,  1.74it/s, Train Loss=0.203] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 50/100: 100%|██████████| 246/246 [02:36<00:00,  1.57it/s, Train Loss=0.0899]


Epoch 50/100, Train Loss: 0.1150


Validating: 100%|██████████| 28/28 [00:10<00:00,  2.56it/s]


Epoch 50/100, Val Loss: 2.7798, Val RMSE: 1.6714, Val PCC: 0.6434, Val R²: 0.3811


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.72it/s]


Test Loss: 2.3612, Test RMSE: 1.5339, Test PCC: 0.4259, Test R²: 0.1197
Epoch 50/100, test RMSE: 1.5339, test PCC: 0.4259, test R²: 0.1197


Epoch 51/100:  29%|██▉       | 72/246 [01:00<01:40,  1.73it/s, Train Loss=0.125] 

Skipping entry 6409 after transform.


Epoch 51/100:  54%|█████▍    | 134/246 [01:39<01:04,  1.74it/s, Train Loss=0.0882]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 51/100: 100%|██████████| 246/246 [02:47<00:00,  1.47it/s, Train Loss=0.122] 


Epoch 51/100, Train Loss: 0.1048


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.12it/s]


Epoch 51/100, Val Loss: 2.7273, Val RMSE: 1.6415, Val PCC: 0.6536, Val R²: 0.4030


Testing: 100%|██████████| 21/21 [00:14<00:00,  1.44it/s]


Test Loss: 2.2225, Test RMSE: 1.4910, Test PCC: 0.4582, Test R²: 0.1681
Epoch 51/100, test RMSE: 1.4910, test PCC: 0.4582, test R²: 0.1681


Epoch 52/100:  57%|█████▋    | 140/246 [01:30<01:03,  1.68it/s, Train Loss=0.089] 

Skipping entry 6409 after transform.


Epoch 52/100:  72%|███████▏  | 177/246 [01:55<00:44,  1.57it/s, Train Loss=0.108] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 52/100: 100%|██████████| 246/246 [02:41<00:00,  1.52it/s, Train Loss=0.0368]


Epoch 52/100, Train Loss: 0.1186


Validating: 100%|██████████| 28/28 [00:15<00:00,  1.84it/s]


Epoch 52/100, Val Loss: 2.5989, Val RMSE: 1.6240, Val PCC: 0.6682, Val R²: 0.4157


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.66it/s]


Test Loss: 2.3364, Test RMSE: 1.5288, Test PCC: 0.4214, Test R²: 0.1255
Epoch 52/100, test RMSE: 1.5288, test PCC: 0.4214, test R²: 0.1255


Epoch 53/100:  49%|████▉     | 121/246 [01:36<01:13,  1.70it/s, Train Loss=0.123] 

Skipping entry 6409 after transform.


Epoch 53/100:  80%|████████  | 197/246 [02:24<00:28,  1.75it/s, Train Loss=0.0959]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 53/100: 100%|██████████| 246/246 [03:00<00:00,  1.37it/s, Train Loss=0.177] 


Epoch 53/100, Train Loss: 0.1148


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.54it/s]


Epoch 53/100, Val Loss: 2.6749, Val RMSE: 1.6306, Val PCC: 0.6598, Val R²: 0.4109


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.23it/s]


Test Loss: 2.2532, Test RMSE: 1.4993, Test PCC: 0.4556, Test R²: 0.1588
Epoch 53/100, test RMSE: 1.4993, test PCC: 0.4556, test R²: 0.1588


Epoch 54/100:  59%|█████▉    | 146/246 [01:30<01:01,  1.64it/s, Train Loss=0.121] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 54/100:  74%|███████▍  | 182/246 [01:55<00:52,  1.21it/s, Train Loss=0.121] 

Skipping entry 6409 after transform.


Epoch 54/100: 100%|██████████| 246/246 [03:02<00:00,  1.35it/s, Train Loss=0.0589]


Epoch 54/100, Train Loss: 0.1315


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.88it/s]


Epoch 54/100, Val Loss: 2.5565, Val RMSE: 1.6162, Val PCC: 0.6645, Val R²: 0.4213


Testing: 100%|██████████| 21/21 [00:15<00:00,  1.38it/s]


Test Loss: 2.3466, Test RMSE: 1.5313, Test PCC: 0.4140, Test R²: 0.1226
Epoch 54/100, test RMSE: 1.5313, test PCC: 0.4140, test R²: 0.1226
Best model saved with Val Loss: 2.5565


Epoch 55/100:  29%|██▉       | 71/246 [01:18<05:12,  1.79s/it, Train Loss=0.126] 

Skipping entry 6409 after transform.


Epoch 55/100:  34%|███▍      | 84/246 [01:33<03:49,  1.42s/it, Train Loss=0.116] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 55/100: 100%|██████████| 246/246 [04:11<00:00,  1.02s/it, Train Loss=0.237] 


Epoch 55/100, Train Loss: 0.1249


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.29it/s]


Epoch 55/100, Val Loss: 3.1330, Val RMSE: 1.6369, Val PCC: 0.6478, Val R²: 0.4063


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.79it/s]


Test Loss: 2.2135, Test RMSE: 1.4893, Test PCC: 0.4508, Test R²: 0.1701
Epoch 55/100, test RMSE: 1.4893, test PCC: 0.4508, test R²: 0.1701


Epoch 56/100:  51%|█████     | 125/246 [01:37<01:44,  1.16it/s, Train Loss=0.0587]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 56/100:  74%|███████▍  | 182/246 [02:18<01:01,  1.03it/s, Train Loss=0.0687]

Skipping entry 6409 after transform.


Epoch 56/100: 100%|██████████| 246/246 [03:04<00:00,  1.33it/s, Train Loss=0.0725]


Epoch 56/100, Train Loss: 0.1112


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.06it/s]


Epoch 56/100, Val Loss: 2.6634, Val RMSE: 1.6386, Val PCC: 0.6554, Val R²: 0.4051


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.90it/s]


Test Loss: 2.2097, Test RMSE: 1.4851, Test PCC: 0.4600, Test R²: 0.1747
Epoch 56/100, test RMSE: 1.4851, test PCC: 0.4600, test R²: 0.1747


Epoch 57/100:  10%|▉         | 24/246 [00:21<02:28,  1.50it/s, Train Loss=0.0696]

Skipping entry 6409 after transform.


Epoch 57/100:  36%|███▌      | 88/246 [01:05<01:44,  1.51it/s, Train Loss=0.0722]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 57/100: 100%|██████████| 246/246 [02:39<00:00,  1.54it/s, Train Loss=0.32]  


Epoch 57/100, Train Loss: 0.0980


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.49it/s]


Epoch 57/100, Val Loss: 2.7116, Val RMSE: 1.6448, Val PCC: 0.6484, Val R²: 0.4006


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.69it/s]


Test Loss: 2.1895, Test RMSE: 1.4810, Test PCC: 0.4555, Test R²: 0.1793
Epoch 57/100, test RMSE: 1.4810, test PCC: 0.4555, test R²: 0.1793


Epoch 58/100:   3%|▎         | 8/246 [00:05<02:26,  1.62it/s, Train Loss=0.0787]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 58/100:  62%|██████▏   | 152/246 [01:30<00:56,  1.67it/s, Train Loss=0.177] 

Skipping entry 6409 after transform.


Epoch 58/100: 100%|██████████| 246/246 [02:31<00:00,  1.63it/s, Train Loss=0.0355]


Epoch 58/100, Train Loss: 0.1299


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.46it/s]


Epoch 58/100, Val Loss: 2.8065, Val RMSE: 1.6909, Val PCC: 0.6374, Val R²: 0.3665


Testing: 100%|██████████| 21/21 [00:10<00:00,  1.99it/s]


Test Loss: 2.3086, Test RMSE: 1.5200, Test PCC: 0.4308, Test R²: 0.1354
Epoch 58/100, test RMSE: 1.5200, test PCC: 0.4308, test R²: 0.1354


Epoch 59/100:  46%|████▌     | 112/246 [01:09<01:18,  1.71it/s, Train Loss=0.0569]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 59/100:  46%|████▋     | 114/246 [01:10<01:16,  1.72it/s, Train Loss=0.0484]

Skipping entry 6409 after transform.


Epoch 59/100: 100%|██████████| 246/246 [02:27<00:00,  1.67it/s, Train Loss=0.171] 


Epoch 59/100, Train Loss: 0.1020


Validating: 100%|██████████| 28/28 [00:20<00:00,  1.36it/s]


Epoch 59/100, Val Loss: 2.6029, Val RMSE: 1.6284, Val PCC: 0.6601, Val R²: 0.4125


Testing: 100%|██████████| 21/21 [00:15<00:00,  1.37it/s]


Test Loss: 2.2585, Test RMSE: 1.5034, Test PCC: 0.4451, Test R²: 0.1543
Epoch 59/100, test RMSE: 1.5034, test PCC: 0.4451, test R²: 0.1543


Epoch 60/100:   3%|▎         | 8/246 [00:07<03:12,  1.24it/s, Train Loss=0.189] 

Skipping entry 6409 after transform.


Epoch 60/100:  54%|█████▍    | 134/246 [01:42<01:08,  1.64it/s, Train Loss=0.0689]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 60/100: 100%|██████████| 246/246 [02:50<00:00,  1.44it/s, Train Loss=0.111] 


Epoch 60/100, Train Loss: 0.0936


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.02it/s]


Epoch 60/100, Val Loss: 2.7407, Val RMSE: 1.6383, Val PCC: 0.6538, Val R²: 0.4053


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.87it/s]


Test Loss: 2.2298, Test RMSE: 1.4940, Test PCC: 0.4451, Test R²: 0.1648
Epoch 60/100, test RMSE: 1.4940, test PCC: 0.4451, test R²: 0.1648


Epoch 61/100:  40%|███▉      | 98/246 [01:06<01:26,  1.71it/s, Train Loss=0.0411]

Skipping entry 6409 after transform.
Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 61/100: 100%|██████████| 246/246 [02:46<00:00,  1.47it/s, Train Loss=0.183] 


Epoch 61/100, Train Loss: 0.0877


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.47it/s]


Epoch 61/100, Val Loss: 2.6653, Val RMSE: 1.6432, Val PCC: 0.6514, Val R²: 0.4018


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.09it/s]


Test Loss: 2.2828, Test RMSE: 1.5112, Test PCC: 0.4328, Test R²: 0.1455
Epoch 61/100, test RMSE: 1.5112, test PCC: 0.4328, test R²: 0.1455


Epoch 62/100:   0%|          | 0/246 [00:00<?, ?it/s]

Skipping entry 6409 after transform.


Epoch 62/100:   8%|▊         | 20/246 [00:13<02:18,  1.63it/s, Train Loss=0.0894]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 62/100: 100%|██████████| 246/246 [02:40<00:00,  1.53it/s, Train Loss=0.0305]


Epoch 62/100, Train Loss: 0.0802


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.96it/s]


Epoch 62/100, Val Loss: 2.6879, Val RMSE: 1.6519, Val PCC: 0.6492, Val R²: 0.3954


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.17it/s]


Test Loss: 2.2413, Test RMSE: 1.4984, Test PCC: 0.4525, Test R²: 0.1599
Epoch 62/100, test RMSE: 1.4984, test PCC: 0.4525, test R²: 0.1599


Epoch 63/100:   2%|▏         | 4/246 [00:04<03:41,  1.09it/s, Train Loss=0.0548]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 63/100:  57%|█████▋    | 139/246 [01:26<01:02,  1.72it/s, Train Loss=0.0597]

Skipping entry 6409 after transform.


Epoch 63/100: 100%|██████████| 246/246 [02:28<00:00,  1.66it/s, Train Loss=0.0425]


Epoch 63/100, Train Loss: 0.0794


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.45it/s]


Epoch 63/100, Val Loss: 2.6028, Val RMSE: 1.6316, Val PCC: 0.6583, Val R²: 0.4102


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.26it/s]


Test Loss: 2.1657, Test RMSE: 1.4730, Test PCC: 0.4631, Test R²: 0.1882
Epoch 63/100, test RMSE: 1.4730, test PCC: 0.4631, test R²: 0.1882


Epoch 64/100:  77%|███████▋  | 190/246 [02:12<00:45,  1.24it/s, Train Loss=0.127] 

Skipping entry 6409 after transform.


Epoch 64/100:  87%|████████▋ | 215/246 [02:31<00:23,  1.33it/s, Train Loss=0.0366]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 64/100: 100%|██████████| 246/246 [02:54<00:00,  1.41it/s, Train Loss=0.102] 


Epoch 64/100, Train Loss: 0.0893


Validating: 100%|██████████| 28/28 [00:19<00:00,  1.45it/s]


Epoch 64/100, Val Loss: 2.5842, Val RMSE: 1.6283, Val PCC: 0.6601, Val R²: 0.4126


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.25it/s]


Test Loss: 2.2404, Test RMSE: 1.4968, Test PCC: 0.4420, Test R²: 0.1617
Epoch 64/100, test RMSE: 1.4968, test PCC: 0.4420, test R²: 0.1617


Epoch 65/100:   7%|▋         | 16/246 [00:17<03:56,  1.03s/it, Train Loss=0.0787]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 65/100:  42%|████▏     | 103/246 [01:35<02:31,  1.06s/it, Train Loss=0.0853]

Skipping entry 6409 after transform.


Epoch 65/100: 100%|██████████| 246/246 [03:01<00:00,  1.35it/s, Train Loss=0.0213]


Epoch 65/100, Train Loss: 0.1010


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.20it/s]


Epoch 65/100, Val Loss: 2.7081, Val RMSE: 1.6366, Val PCC: 0.6570, Val R²: 0.4066


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.81it/s]


Test Loss: 2.1975, Test RMSE: 1.4816, Test PCC: 0.4587, Test R²: 0.1787
Epoch 65/100, test RMSE: 1.4816, test PCC: 0.4587, test R²: 0.1787


Epoch 66/100:  82%|████████▏ | 202/246 [02:23<00:27,  1.60it/s, Train Loss=0.163] 

Skipping entry 6409 after transform.


Epoch 66/100:  88%|████████▊ | 216/246 [02:33<00:22,  1.32it/s, Train Loss=0.148] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 66/100: 100%|██████████| 246/246 [02:54<00:00,  1.41it/s, Train Loss=0.0334]


Epoch 66/100, Train Loss: 0.1126


Validating: 100%|██████████| 28/28 [00:16<00:00,  1.68it/s]


Epoch 66/100, Val Loss: 2.6665, Val RMSE: 1.6493, Val PCC: 0.6506, Val R²: 0.3973


Testing: 100%|██████████| 21/21 [00:16<00:00,  1.28it/s]


Test Loss: 2.2558, Test RMSE: 1.5034, Test PCC: 0.4395, Test R²: 0.1542
Epoch 66/100, test RMSE: 1.5034, test PCC: 0.4395, test R²: 0.1542


Epoch 67/100:  55%|█████▌    | 136/246 [01:46<01:04,  1.70it/s, Train Loss=0.0954]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 67/100:  81%|████████▏ | 200/246 [02:32<00:31,  1.47it/s, Train Loss=0.0601]

Skipping entry 6409 after transform.


Epoch 67/100: 100%|██████████| 246/246 [02:59<00:00,  1.37it/s, Train Loss=0.0198]


Epoch 67/100, Train Loss: 0.1000


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.50it/s]


Epoch 67/100, Val Loss: 2.7359, Val RMSE: 1.6691, Val PCC: 0.6416, Val R²: 0.3828


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.21it/s]


Test Loss: 2.3165, Test RMSE: 1.5210, Test PCC: 0.4291, Test R²: 0.1344
Epoch 67/100, test RMSE: 1.5210, test PCC: 0.4291, test R²: 0.1344


Epoch 68/100:   1%|          | 2/246 [00:02<04:58,  1.22s/it, Train Loss=0.234] 

Skipping entry 6409 after transform.


Epoch 68/100:  76%|███████▌  | 186/246 [02:22<00:40,  1.48it/s, Train Loss=0.194] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 68/100: 100%|██████████| 246/246 [02:58<00:00,  1.38it/s, Train Loss=0.12]  


Epoch 68/100, Train Loss: 0.1006


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.30it/s]


Epoch 68/100, Val Loss: 2.6020, Val RMSE: 1.6349, Val PCC: 0.6581, Val R²: 0.4078


Testing: 100%|██████████| 21/21 [00:14<00:00,  1.44it/s]


Test Loss: 2.1981, Test RMSE: 1.4832, Test PCC: 0.4578, Test R²: 0.1769
Epoch 68/100, test RMSE: 1.4832, test PCC: 0.4578, test R²: 0.1769


Epoch 69/100:  11%|█         | 27/246 [00:32<05:07,  1.41s/it, Train Loss=0.0583]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 69/100:  45%|████▍     | 110/246 [01:26<01:18,  1.74it/s, Train Loss=0.0965]

Skipping entry 6409 after transform.


Epoch 69/100: 100%|██████████| 246/246 [02:52<00:00,  1.43it/s, Train Loss=0.0455]


Epoch 69/100, Train Loss: 0.0995


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.22it/s]


Epoch 69/100, Val Loss: 3.1433, Val RMSE: 1.6383, Val PCC: 0.6571, Val R²: 0.4053


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.31it/s]


Test Loss: 2.3002, Test RMSE: 1.5176, Test PCC: 0.4301, Test R²: 0.1382
Epoch 69/100, test RMSE: 1.5176, test PCC: 0.4301, test R²: 0.1382


Epoch 70/100:   0%|          | 0/246 [00:00<?, ?it/s]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 70/100:   4%|▍         | 11/246 [00:07<02:21,  1.66it/s, Train Loss=0.0773]

Skipping entry 6409 after transform.


Epoch 70/100: 100%|██████████| 246/246 [02:25<00:00,  1.69it/s, Train Loss=0.182] 


Epoch 70/100, Train Loss: 0.0866


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.37it/s]


Epoch 70/100, Val Loss: 2.6098, Val RMSE: 1.6310, Val PCC: 0.6586, Val R²: 0.4106


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.04it/s]


Test Loss: 2.2786, Test RMSE: 1.5095, Test PCC: 0.4364, Test R²: 0.1474
Epoch 70/100, test RMSE: 1.5095, test PCC: 0.4364, test R²: 0.1474


Epoch 71/100:  16%|█▌        | 39/246 [00:30<02:02,  1.69it/s, Train Loss=0.208] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 71/100:  40%|████      | 99/246 [01:06<01:38,  1.49it/s, Train Loss=0.0909]

Skipping entry 6409 after transform.


Epoch 71/100: 100%|██████████| 246/246 [02:42<00:00,  1.51it/s, Train Loss=0.00632]


Epoch 71/100, Train Loss: 0.1037


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.09it/s]


Epoch 71/100, Val Loss: 3.2635, Val RMSE: 1.6355, Val PCC: 0.6580, Val R²: 0.4073


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.77it/s]


Test Loss: 2.2852, Test RMSE: 1.5102, Test PCC: 0.4347, Test R²: 0.1466
Epoch 71/100, test RMSE: 1.5102, test PCC: 0.4347, test R²: 0.1466


Epoch 72/100:   5%|▍         | 12/246 [00:09<02:32,  1.53it/s, Train Loss=0.0358]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 72/100:  38%|███▊      | 94/246 [01:05<01:52,  1.35it/s, Train Loss=0.0748]

Skipping entry 6409 after transform.


Epoch 72/100: 100%|██████████| 246/246 [02:40<00:00,  1.53it/s, Train Loss=0.0756]


Epoch 72/100, Train Loss: 0.0780


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.18it/s]


Epoch 72/100, Val Loss: 2.6206, Val RMSE: 1.6311, Val PCC: 0.6554, Val R²: 0.4105


Testing: 100%|██████████| 21/21 [00:10<00:00,  1.99it/s]


Test Loss: 2.3156, Test RMSE: 1.5210, Test PCC: 0.4280, Test R²: 0.1344
Epoch 72/100, test RMSE: 1.5210, test PCC: 0.4280, test R²: 0.1344


Epoch 73/100:  52%|█████▏    | 129/246 [01:51<01:14,  1.58it/s, Train Loss=0.065] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 73/100:  85%|████████▍ | 209/246 [02:46<00:28,  1.30it/s, Train Loss=0.043] 

Skipping entry 6409 after transform.


Epoch 73/100: 100%|██████████| 246/246 [03:08<00:00,  1.30it/s, Train Loss=0.031] 


Epoch 73/100, Train Loss: 0.0812


Validating: 100%|██████████| 28/28 [00:15<00:00,  1.84it/s]


Epoch 73/100, Val Loss: 2.5407, Val RMSE: 1.6058, Val PCC: 0.6704, Val R²: 0.4287


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.91it/s]


Test Loss: 2.2486, Test RMSE: 1.4970, Test PCC: 0.4412, Test R²: 0.1615
Epoch 73/100, test RMSE: 1.4970, test PCC: 0.4412, test R²: 0.1615
Best model saved with Val Loss: 2.5407


Epoch 74/100:  26%|██▋       | 65/246 [00:44<01:46,  1.70it/s, Train Loss=0.0759]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 74/100:  89%|████████▉ | 219/246 [02:36<00:16,  1.67it/s, Train Loss=0.0301]

Skipping entry 6409 after transform.


Epoch 74/100: 100%|██████████| 246/246 [02:55<00:00,  1.40it/s, Train Loss=0.057] 


Epoch 74/100, Train Loss: 0.0914


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.16it/s]


Epoch 74/100, Val Loss: 2.5386, Val RMSE: 1.6142, Val PCC: 0.6712, Val R²: 0.4227


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.17it/s]


Test Loss: 2.2278, Test RMSE: 1.4945, Test PCC: 0.4613, Test R²: 0.1642
Epoch 74/100, test RMSE: 1.4945, test PCC: 0.4613, test R²: 0.1642
Best model saved with Val Loss: 2.5386


Epoch 75/100:  47%|████▋     | 115/246 [01:26<01:40,  1.30it/s, Train Loss=0.0929]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 75/100:  67%|██████▋   | 166/246 [02:00<00:48,  1.66it/s, Train Loss=0.0456]

Skipping entry 6409 after transform.


Epoch 75/100: 100%|██████████| 246/246 [02:49<00:00,  1.45it/s, Train Loss=0.00988]


Epoch 75/100, Train Loss: 0.0849


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.13it/s]


Epoch 75/100, Val Loss: 3.2384, Val RMSE: 1.6166, Val PCC: 0.6632, Val R²: 0.4210


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.90it/s]


Test Loss: 2.3156, Test RMSE: 1.5222, Test PCC: 0.4252, Test R²: 0.1330
Epoch 75/100, test RMSE: 1.5222, test PCC: 0.4252, test R²: 0.1330


Epoch 76/100:  56%|█████▌    | 137/246 [01:40<01:06,  1.63it/s, Train Loss=0.0479]

Skipping entry 6409 after transform.


Epoch 76/100:  73%|███████▎  | 179/246 [02:12<01:07,  1.01s/it, Train Loss=0.0306]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 76/100: 100%|██████████| 246/246 [03:02<00:00,  1.35it/s, Train Loss=0.0564]


Epoch 76/100, Train Loss: 0.0762


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.36it/s]


Epoch 76/100, Val Loss: 2.5831, Val RMSE: 1.6181, Val PCC: 0.6637, Val R²: 0.4199


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.73it/s]


Test Loss: 2.2811, Test RMSE: 1.5104, Test PCC: 0.4314, Test R²: 0.1464
Epoch 76/100, test RMSE: 1.5104, test PCC: 0.4314, test R²: 0.1464


Epoch 77/100:  19%|█▊        | 46/246 [00:31<02:05,  1.60it/s, Train Loss=0.0515]

Skipping entry 6409 after transform.


Epoch 77/100:  36%|███▌      | 89/246 [00:59<01:31,  1.71it/s, Train Loss=0.0572]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 77/100: 100%|██████████| 246/246 [02:30<00:00,  1.64it/s, Train Loss=0.0712]


Epoch 77/100, Train Loss: 0.0674


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.10it/s]


Epoch 77/100, Val Loss: 2.5854, Val RMSE: 1.6120, Val PCC: 0.6666, Val R²: 0.4243


Testing: 100%|██████████| 21/21 [00:13<00:00,  1.58it/s]


Test Loss: 2.1991, Test RMSE: 1.4832, Test PCC: 0.4571, Test R²: 0.1769
Epoch 77/100, test RMSE: 1.4832, test PCC: 0.4571, test R²: 0.1769


Epoch 78/100:  80%|████████  | 197/246 [02:19<00:29,  1.67it/s, Train Loss=0.057] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 78/100:  83%|████████▎ | 204/246 [02:23<00:25,  1.63it/s, Train Loss=0.0538]

Skipping entry 6409 after transform.


Epoch 78/100: 100%|██████████| 246/246 [02:48<00:00,  1.46it/s, Train Loss=0.00683]


Epoch 78/100, Train Loss: 0.0628


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.94it/s]


Epoch 78/100, Val Loss: 3.4119, Val RMSE: 1.6177, Val PCC: 0.6626, Val R²: 0.4202


Testing: 100%|██████████| 21/21 [00:16<00:00,  1.25it/s]


Test Loss: 2.2666, Test RMSE: 1.5048, Test PCC: 0.4334, Test R²: 0.1527
Epoch 78/100, test RMSE: 1.5048, test PCC: 0.4334, test R²: 0.1527


Epoch 79/100:  77%|███████▋  | 189/246 [01:58<00:32,  1.77it/s, Train Loss=0.0538]

Skipping entry 6409 after transform.


Epoch 79/100:  77%|███████▋  | 190/246 [01:59<00:31,  1.76it/s, Train Loss=0.0657]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 79/100: 100%|██████████| 246/246 [02:35<00:00,  1.58it/s, Train Loss=0.0289]


Epoch 79/100, Train Loss: 0.0628


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.07it/s]


Epoch 79/100, Val Loss: 2.5192, Val RMSE: 1.5951, Val PCC: 0.6738, Val R²: 0.4362


Testing: 100%|██████████| 21/21 [00:20<00:00,  1.03it/s]


Test Loss: 2.2273, Test RMSE: 1.4927, Test PCC: 0.4473, Test R²: 0.1662
Epoch 79/100, test RMSE: 1.4927, test PCC: 0.4473, test R²: 0.1662
Best model saved with Val Loss: 2.5192


Epoch 80/100:  76%|███████▋  | 188/246 [02:32<00:51,  1.12it/s, Train Loss=0.0604]

Skipping entry 6409 after transform.


Epoch 80/100:  90%|█████████ | 222/246 [02:57<00:21,  1.13it/s, Train Loss=0.0781]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 80/100: 100%|██████████| 246/246 [03:16<00:00,  1.25it/s, Train Loss=0.0464]


Epoch 80/100, Train Loss: 0.0700


Validating: 100%|██████████| 28/28 [00:15<00:00,  1.84it/s]


Epoch 80/100, Val Loss: 2.5756, Val RMSE: 1.6243, Val PCC: 0.6590, Val R²: 0.4155


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.87it/s]


Test Loss: 2.2147, Test RMSE: 1.4884, Test PCC: 0.4463, Test R²: 0.1711
Epoch 80/100, test RMSE: 1.4884, test PCC: 0.4463, test R²: 0.1711


Epoch 81/100:  19%|█▊        | 46/246 [00:44<02:46,  1.20it/s, Train Loss=0.0578]

Skipping entry 6409 after transform.


Epoch 81/100:  30%|███       | 75/246 [01:06<02:25,  1.17it/s, Train Loss=0.0603]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 81/100: 100%|██████████| 246/246 [03:00<00:00,  1.36it/s, Train Loss=0.0828]


Epoch 81/100, Train Loss: 0.0732


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.26it/s]


Epoch 81/100, Val Loss: 2.5553, Val RMSE: 1.6116, Val PCC: 0.6709, Val R²: 0.4245


Testing: 100%|██████████| 21/21 [00:08<00:00,  2.41it/s]


Test Loss: 2.2068, Test RMSE: 1.4859, Test PCC: 0.4565, Test R²: 0.1739
Epoch 81/100, test RMSE: 1.4859, test PCC: 0.4565, test R²: 0.1739


Epoch 82/100:  91%|█████████ | 224/246 [02:31<00:15,  1.45it/s, Train Loss=0.243] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 82/100:  93%|█████████▎| 230/246 [02:36<00:11,  1.36it/s, Train Loss=0.0379]

Skipping entry 6409 after transform.


Epoch 82/100: 100%|██████████| 246/246 [02:45<00:00,  1.48it/s, Train Loss=0.0349]


Epoch 82/100, Train Loss: 0.0770


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.19it/s]


Epoch 82/100, Val Loss: 2.5873, Val RMSE: 1.6112, Val PCC: 0.6628, Val R²: 0.4248


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.23it/s]


Test Loss: 2.1859, Test RMSE: 1.4777, Test PCC: 0.4583, Test R²: 0.1830
Epoch 82/100, test RMSE: 1.4777, test PCC: 0.4583, test R²: 0.1830


Epoch 83/100:  59%|█████▉    | 145/246 [01:36<01:03,  1.59it/s, Train Loss=0.101] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 83/100:  94%|█████████▍| 231/246 [02:32<00:09,  1.66it/s, Train Loss=0.0704]

Skipping entry 6409 after transform.


Epoch 83/100: 100%|██████████| 246/246 [02:41<00:00,  1.53it/s, Train Loss=0.0962]


Epoch 83/100, Train Loss: 0.0943


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.91it/s]


Epoch 83/100, Val Loss: 2.4458, Val RMSE: 1.5820, Val PCC: 0.6796, Val R²: 0.4455


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.85it/s]


Test Loss: 2.1918, Test RMSE: 1.4790, Test PCC: 0.4550, Test R²: 0.1815
Epoch 83/100, test RMSE: 1.4790, test PCC: 0.4550, test R²: 0.1815
Best model saved with Val Loss: 2.4458


Epoch 84/100:  30%|███       | 74/246 [00:49<01:44,  1.64it/s, Train Loss=0.092] 

Skipping entry 6409 after transform.


Epoch 84/100:  37%|███▋      | 90/246 [00:59<01:57,  1.33it/s, Train Loss=0.0686]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 84/100: 100%|██████████| 246/246 [02:29<00:00,  1.65it/s, Train Loss=0.0319]


Epoch 84/100, Train Loss: 0.1086


Validating: 100%|██████████| 28/28 [00:13<00:00,  2.08it/s]


Epoch 84/100, Val Loss: 2.5738, Val RMSE: 1.6117, Val PCC: 0.6650, Val R²: 0.4245


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.19it/s]


Test Loss: 2.1815, Test RMSE: 1.4775, Test PCC: 0.4630, Test R²: 0.1832
Epoch 84/100, test RMSE: 1.4775, test PCC: 0.4630, test R²: 0.1832


Epoch 85/100:  42%|████▏     | 104/246 [01:00<01:20,  1.77it/s, Train Loss=0.0645]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 85/100:  49%|████▉     | 120/246 [01:09<01:11,  1.77it/s, Train Loss=0.0704]

Skipping entry 6409 after transform.


Epoch 85/100: 100%|██████████| 246/246 [02:21<00:00,  1.74it/s, Train Loss=0.0729]


Epoch 85/100, Train Loss: 0.0935


Validating: 100%|██████████| 28/28 [00:10<00:00,  2.57it/s]


Epoch 85/100, Val Loss: 2.4780, Val RMSE: 1.5904, Val PCC: 0.6768, Val R²: 0.4396


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.08it/s]


Test Loss: 2.2472, Test RMSE: 1.5008, Test PCC: 0.4389, Test R²: 0.1572
Epoch 85/100, test RMSE: 1.5008, test PCC: 0.4389, test R²: 0.1572


Epoch 86/100:  16%|█▌        | 39/246 [00:26<02:07,  1.63it/s, Train Loss=0.0466]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 86/100:  81%|████████▏ | 200/246 [02:02<00:27,  1.64it/s, Train Loss=0.0252]

Skipping entry 6409 after transform.


Epoch 86/100: 100%|██████████| 246/246 [02:29<00:00,  1.65it/s, Train Loss=0.0131]


Epoch 86/100, Train Loss: 0.0708


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.46it/s]


Epoch 86/100, Val Loss: 2.4502, Val RMSE: 1.5831, Val PCC: 0.6817, Val R²: 0.4448


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.30it/s]


Test Loss: 2.2087, Test RMSE: 1.4859, Test PCC: 0.4562, Test R²: 0.1739
Epoch 86/100, test RMSE: 1.4859, test PCC: 0.4562, test R²: 0.1739


Epoch 87/100:   0%|          | 0/246 [00:00<?, ?it/s]

Skipping entry 6409 after transform.


Epoch 87/100:  27%|██▋       | 66/246 [00:40<01:46,  1.69it/s, Train Loss=0.0148]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 87/100: 100%|██████████| 246/246 [02:25<00:00,  1.70it/s, Train Loss=0.109] 


Epoch 87/100, Train Loss: 0.0612


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.24it/s]


Epoch 87/100, Val Loss: 2.4955, Val RMSE: 1.5858, Val PCC: 0.6804, Val R²: 0.4428


Testing: 100%|██████████| 21/21 [00:12<00:00,  1.72it/s]


Test Loss: 2.1912, Test RMSE: 1.4822, Test PCC: 0.4561, Test R²: 0.1780
Epoch 87/100, test RMSE: 1.4822, test PCC: 0.4561, test R²: 0.1780


Epoch 88/100:   1%|          | 2/246 [00:02<05:04,  1.25s/it, Train Loss=0.0663]

Skipping entry 6409 after transform.


Epoch 88/100:   1%|          | 3/246 [00:03<03:48,  1.07it/s, Train Loss=0.107] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 88/100: 100%|██████████| 246/246 [02:26<00:00,  1.67it/s, Train Loss=0.261] 


Epoch 88/100, Train Loss: 0.0873


Validating: 100%|██████████| 28/28 [00:16<00:00,  1.70it/s]


Epoch 88/100, Val Loss: 2.4888, Val RMSE: 1.5944, Val PCC: 0.6722, Val R²: 0.4368


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.15it/s]


Test Loss: 2.1643, Test RMSE: 1.4723, Test PCC: 0.4630, Test R²: 0.1889
Epoch 88/100, test RMSE: 1.4723, test PCC: 0.4630, test R²: 0.1889


Epoch 89/100:  33%|███▎      | 80/246 [00:49<01:34,  1.76it/s, Train Loss=0.0725]

Skipping entry 6409 after transform.


Epoch 89/100:  41%|████      | 101/246 [01:01<01:22,  1.77it/s, Train Loss=0.0702]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 89/100: 100%|██████████| 246/246 [02:23<00:00,  1.72it/s, Train Loss=0.104] 


Epoch 89/100, Train Loss: 0.0702


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.18it/s]


Epoch 89/100, Val Loss: 2.4967, Val RMSE: 1.5968, Val PCC: 0.6766, Val R²: 0.4351


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.86it/s]


Test Loss: 2.1795, Test RMSE: 1.4778, Test PCC: 0.4652, Test R²: 0.1828
Epoch 89/100, test RMSE: 1.4778, test PCC: 0.4652, test R²: 0.1828


Epoch 90/100:  41%|████      | 100/246 [01:04<01:27,  1.66it/s, Train Loss=0.055]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 90/100:  50%|████▉     | 122/246 [01:18<01:26,  1.43it/s, Train Loss=0.0317]

Skipping entry 6409 after transform.


Epoch 90/100: 100%|██████████| 246/246 [02:36<00:00,  1.58it/s, Train Loss=0.0783]


Epoch 90/100, Train Loss: 0.0552


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.22it/s]


Epoch 90/100, Val Loss: 2.4404, Val RMSE: 1.5811, Val PCC: 0.6796, Val R²: 0.4461


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.16it/s]


Test Loss: 2.1646, Test RMSE: 1.4721, Test PCC: 0.4664, Test R²: 0.1891
Epoch 90/100, test RMSE: 1.4721, test PCC: 0.4664, test R²: 0.1891
Best model saved with Val Loss: 2.4404


Epoch 91/100:   9%|▉         | 23/246 [00:16<02:13,  1.68it/s, Train Loss=0.0402]

Skipping entry 6409 after transform.


Epoch 91/100:  70%|███████   | 173/246 [01:42<00:41,  1.75it/s, Train Loss=0.0517]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 91/100: 100%|██████████| 246/246 [02:24<00:00,  1.71it/s, Train Loss=0.0563]


Epoch 91/100, Train Loss: 0.0516


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.37it/s]


Epoch 91/100, Val Loss: 2.4698, Val RMSE: 1.5842, Val PCC: 0.6763, Val R²: 0.4439


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.12it/s]


Test Loss: 2.1341, Test RMSE: 1.4609, Test PCC: 0.4753, Test R²: 0.2014
Epoch 91/100, test RMSE: 1.4609, test PCC: 0.4753, test R²: 0.2014


Epoch 92/100:  58%|█████▊    | 142/246 [01:27<01:01,  1.70it/s, Train Loss=0.0348]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 92/100:  59%|█████▉    | 145/246 [01:29<00:58,  1.73it/s, Train Loss=0.0384]

Skipping entry 6409 after transform.


Epoch 92/100: 100%|██████████| 246/246 [02:28<00:00,  1.65it/s, Train Loss=0.0653]


Epoch 92/100, Train Loss: 0.0545


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.97it/s]


Epoch 92/100, Val Loss: 2.5040, Val RMSE: 1.6019, Val PCC: 0.6664, Val R²: 0.4314


Testing: 100%|██████████| 21/21 [00:10<00:00,  1.98it/s]


Test Loss: 2.1117, Test RMSE: 1.4534, Test PCC: 0.4824, Test R²: 0.2096
Epoch 92/100, test RMSE: 1.4534, test PCC: 0.4824, test R²: 0.2096


Epoch 93/100:  35%|███▍      | 85/246 [00:58<01:44,  1.54it/s, Train Loss=0.0271]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 93/100:  74%|███████▍  | 183/246 [01:59<00:36,  1.74it/s, Train Loss=0.0765]

Skipping entry 6409 after transform.


Epoch 93/100: 100%|██████████| 246/246 [02:37<00:00,  1.57it/s, Train Loss=0.0254]


Epoch 93/100, Train Loss: 0.0597


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.91it/s]


Epoch 93/100, Val Loss: 2.4484, Val RMSE: 1.5648, Val PCC: 0.6876, Val R²: 0.4575


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.14it/s]


Test Loss: 2.1723, Test RMSE: 1.4719, Test PCC: 0.4654, Test R²: 0.1894
Epoch 93/100, test RMSE: 1.4719, test PCC: 0.4654, test R²: 0.1894


Epoch 94/100:  20%|█▉        | 48/246 [00:36<01:57,  1.68it/s, Train Loss=0.0466]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 94/100:  59%|█████▊    | 144/246 [01:32<00:59,  1.70it/s, Train Loss=0.0701]

Skipping entry 6409 after transform.


Epoch 94/100: 100%|██████████| 246/246 [02:33<00:00,  1.60it/s, Train Loss=0.135] 


Epoch 94/100, Train Loss: 0.0582


Validating: 100%|██████████| 28/28 [00:14<00:00,  1.94it/s]


Epoch 94/100, Val Loss: 2.4032, Val RMSE: 1.5594, Val PCC: 0.6872, Val R²: 0.4612


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.32it/s]


Test Loss: 2.1624, Test RMSE: 1.4697, Test PCC: 0.4663, Test R²: 0.1918
Epoch 94/100, test RMSE: 1.4697, test PCC: 0.4663, test R²: 0.1918
Best model saved with Val Loss: 2.4032


Epoch 95/100:  41%|████▏     | 102/246 [01:04<01:22,  1.74it/s, Train Loss=0.0335]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 95/100:  95%|█████████▍| 233/246 [02:21<00:11,  1.10it/s, Train Loss=0.0572]

Skipping entry 6409 after transform.


Epoch 95/100: 100%|██████████| 246/246 [02:29<00:00,  1.65it/s, Train Loss=0.0353]


Epoch 95/100, Train Loss: 0.0683


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.30it/s]


Epoch 95/100, Val Loss: 2.5047, Val RMSE: 1.6015, Val PCC: 0.6754, Val R²: 0.4317


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.81it/s]


Test Loss: 2.1769, Test RMSE: 1.4773, Test PCC: 0.4621, Test R²: 0.1834
Epoch 95/100, test RMSE: 1.4773, test PCC: 0.4621, test R²: 0.1834


Epoch 96/100:   6%|▌         | 14/246 [00:11<02:16,  1.70it/s, Train Loss=0.131] 

Skipping entry 6409 after transform.


Epoch 96/100:  77%|███████▋  | 190/246 [01:54<00:34,  1.60it/s, Train Loss=0.0729]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 96/100: 100%|██████████| 246/246 [02:28<00:00,  1.65it/s, Train Loss=0.00889]


Epoch 96/100, Train Loss: 0.0636


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.33it/s]


Epoch 96/100, Val Loss: 2.5130, Val RMSE: 1.5986, Val PCC: 0.6762, Val R²: 0.4338


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.80it/s]


Test Loss: 2.2233, Test RMSE: 1.4884, Test PCC: 0.4510, Test R²: 0.1711
Epoch 96/100, test RMSE: 1.4884, test PCC: 0.4510, test R²: 0.1711


Epoch 97/100:   5%|▌         | 13/246 [00:11<02:56,  1.32it/s, Train Loss=0.0614]

Skipping entry 6409 after transform.


Epoch 97/100:  53%|█████▎    | 130/246 [01:26<01:06,  1.74it/s, Train Loss=0.102] 

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 97/100: 100%|██████████| 246/246 [02:42<00:00,  1.51it/s, Train Loss=0.076] 


Epoch 97/100, Train Loss: 0.0551


Validating: 100%|██████████| 28/28 [00:11<00:00,  2.45it/s]


Epoch 97/100, Val Loss: 2.4594, Val RMSE: 1.5784, Val PCC: 0.6847, Val R²: 0.4480


Testing: 100%|██████████| 21/21 [00:10<00:00,  2.00it/s]


Test Loss: 2.2012, Test RMSE: 1.4839, Test PCC: 0.4600, Test R²: 0.1760
Epoch 97/100, test RMSE: 1.4839, test PCC: 0.4600, test R²: 0.1760


Epoch 98/100:  18%|█▊        | 44/246 [00:29<02:02,  1.66it/s, Train Loss=0.0579]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 98/100:  71%|███████   | 175/246 [01:48<00:40,  1.74it/s, Train Loss=0.063] 

Skipping entry 6409 after transform.


Epoch 98/100: 100%|██████████| 246/246 [02:31<00:00,  1.62it/s, Train Loss=0.0505]


Epoch 98/100, Train Loss: 0.0570


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.33it/s]


Epoch 98/100, Val Loss: 2.4739, Val RMSE: 1.5900, Val PCC: 0.6748, Val R²: 0.4399


Testing: 100%|██████████| 21/21 [00:11<00:00,  1.87it/s]


Test Loss: 2.1686, Test RMSE: 1.4731, Test PCC: 0.4602, Test R²: 0.1880
Epoch 98/100, test RMSE: 1.4731, test PCC: 0.4602, test R²: 0.1880


Epoch 99/100:  21%|██        | 52/246 [00:34<01:54,  1.70it/s, Train Loss=0.0372]

Skipping entry 6409 after transform.

Epoch 99/100:  22%|██▏       | 53/246 [00:34<01:53,  1.71it/s, Train Loss=0.0372]

Epoch 99/100:  91%|█████████ | 224/246 [02:15<00:15,  1.46it/s, Train Loss=0.0618]

Skipping mutation K->S at ('A', '80'): not found in seq_map


Epoch 99/100:  91%|█████████▏| 225/246 [02:16<00:13,  1.53it/s, Train Loss=0.0465]

Skipping entry 1703 after transform.


Epoch 99/100: 100%|██████████| 246/246 [02:28<00:00,  1.65it/s, Train Loss=0.132] 


Epoch 99/100, Train Loss: 0.0534


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.21it/s]


Epoch 99/100, Val Loss: 2.5332, Val RMSE: 1.5973, Val PCC: 0.6699, Val R²: 0.4347


Testing: 100%|██████████| 21/21 [00:09<00:00,  2.12it/s]


Test Loss: 2.1591, Test RMSE: 1.4696, Test PCC: 0.4663, Test R²: 0.1918
Epoch 99/100, test RMSE: 1.4696, test PCC: 0.4663, test R²: 0.1918


Epoch 100/100:  20%|█▉        | 48/246 [00:33<01:53,  1.74it/s, Train Loss=0.0485]

Skipping mutation K->S at ('A', '80'): not found in seq_map
Skipping entry 1703 after transform.


Epoch 100/100:  94%|█████████▍| 232/246 [02:21<00:07,  1.75it/s, Train Loss=0.0913]

Skipping entry 6409 after transform.


Epoch 100/100: 100%|██████████| 246/246 [02:30<00:00,  1.63it/s, Train Loss=0.0307]


Epoch 100/100, Train Loss: 0.0760


Validating: 100%|██████████| 28/28 [00:12<00:00,  2.25it/s]


Epoch 100/100, Val Loss: 2.4270, Val RMSE: 1.5765, Val PCC: 0.6858, Val R²: 0.4493


Testing: 100%|██████████| 21/21 [00:13<00:00,  1.59it/s]


Test Loss: 2.3641, Test RMSE: 1.5358, Test PCC: 0.4085, Test R²: 0.1175
Epoch 100/100, test RMSE: 1.5358, test PCC: 0.4085, test R²: 0.1175


In [15]:
best_model_path = "/home/hongtan/pretrain_single/weight/best_model.pth"
model.to(device)
model.load_state_dict(torch.load(best_model_path))

# 在测试集上测试
test_model(model, test_loader, device='cuda:0')

/tmp/ipykernel_1272984/206737171.py:3: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  model.load_state_dict(torch.load(best_model_path))
Testing: 100%|██████████| 21/21 [00:0

Test Loss: 2.1639, Test RMSE: 1.4721, Test PCC: 0.4641, Test R²: 0.1891


(1.4720819, 0.46406485578096124, 0.18914538621902466, 2.163921418644133)

Epoch 1/1000, Train Loss: 3.7755478318733506
Validating: 100%|██████████| 9/9 [00:15<00:00,  1.73s/it]
Validation Loss: 3.440200130144755
Epoch 1/1000, Val RMSE: 1.8626, Val PCC: 0.6669, Val R²: 0.4344
Epoch 2/1000, Train Loss: 2.769809875307204
Validating: 100%|██████████| 9/9 [00:18<00:00,  2.03s/it]
Validation Loss: 3.4820684327019586
Epoch 2/1000, Val RMSE: 1.8031, Val PCC: 0.6885, Val R²: 0.4700
Epoch 3/1000, Train Loss: 2.436043185523794
Validating: 100%|██████████| 9/9 [00:14<00:00,  1.66s/it]
Validation Loss: 3.3180822796291776
Epoch 3/1000, Val RMSE: 1.8264, Val PCC: 0.6820, Val R²: 0.4562
Epoch 4/1000, Train Loss: 2.396531761447086
Validating: 100%|██████████| 9/9 [00:19<00:00,  2.22s/it]
Validation Loss: 3.402861820326911
Epoch 4/1000, Val RMSE: 1.8588, Val PCC: 0.7065, Val R²: 0.4368
Epoch 5/1000, Train Loss: 2.2285565168042725
Validating: 100%|██████████| 9/9 [00:17<00:00,  1.97s/it]
Validation Loss: 3.3189518451690674
Epoch 5/1000, Val RMSE: 1.8318, Val PCC: 0.6858, Val R²: 0.4530
Epoch 6/1000, Train Loss: 2.437140565884264
Validating: 100%|██████████| 9/9 [00:17<00:00,  1.94s/it]
Validation Loss: 3.866072667969598
Epoch 6/1000, Val RMSE: 1.9175, Val PCC: 0.6353, Val R²: 0.4006
Epoch 7/1000, Train Loss: 2.5104055570650705
Validating: 100%|██████████| 9/9 [00:16<00:00,  1.81s/it]
Validation Loss: 4.321215364668104
Epoch 7/1000, Val RMSE: 2.0865, Val PCC: 0.5843, Val R²: 0.2903
Epoch 8/1000, Train Loss: 4.1250943533981905
Validating: 100%|██████████| 9/9 [00:16<00:00,  1.83s/it]
Validation Loss: 4.389447000291613
Epoch 8/1000, Val RMSE: 2.0988, Val PCC: 0.5502, Val R²: 0.2819
Epoch 9/1000, Train Loss: 3.9547084675559514
Validating: 100%|██████████| 9/9 [00:17<00:00,  1.98s/it]
Validation Loss: 4.163823180728489
Epoch 9/1000, Val RMSE: 2.0451, Val PCC: 0.5684, Val R²: 0.3181